{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Comparing mutational effects among Env homologs\n",
    "\n",
    "## Overview\n",
    "This Jupyter notebook compares the effects of mutations between two homologs of HIV Env, the BG505 and BF520 strains.\n",
    "The goal is to quantify how much the effects of mutations have shifted between the viral strains.\n",
    "\n",
    "This analysis was done by Hugh Haddox, Adam Dingens, Sarah Hilton, and Jesse Bloom."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import modules and define common paths\n",
    "First we import Python modules used throughout the notebook, and define key directory paths."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using dms_tools2 version 2.2.dev0\n",
      "Using phydms version 2.1.3\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import string\n",
    "import itertools\n",
    "import multiprocessing\n",
    "import pandas\n",
    "import Bio.SeqIO\n",
    "import Bio.Seq\n",
    "from IPython.display import display, HTML\n",
    "import dms_tools2\n",
    "print(\"Using dms_tools2 version {0}\".format(dms_tools2.__version__))\n",
    "from dms_tools2.ipython_utils import showPDF\n",
    "import dms_tools2.sra\n",
    "import phydmslib\n",
    "print(\"Using phydms version {0}\".format(phydmslib.__version__))\n",
    "\n",
    "# define directories in which results are placed\n",
    "resultsdir = './results/'\n",
    "if not os.path.isdir(resultsdir):\n",
    "    os.mkdir(resultsdir)\n",
    "    \n",
    "# do we use existing program output if already present?\n",
    "use_existing = 'yes'\n",
    "\n",
    "# Use up to 6 CPUs, more sometimes causes memory problem on Hutch server\n",
    "ncpus = min(6, multiprocessing.cpu_count())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Specify Env coding sequences, homolog alignments, and site numbering\n",
    "A major complexity is how to number and align the two Env homologs.\n",
    "For computational purposes, it is often convenient to number the sequences sequentially as 1, 2, ... beginning with the N-terminal methionine.\n",
    "However, the standard numbering scheme for HIV Env is the [HXB2 numbering scheme](https://www.hiv.lanl.gov/content/sequence/HIV/REVIEWS/HXB2.html).\n",
    "In this notebook, we process the FASTQ files and count the mutations in sequential numbering, and then map these numbers to the [HXB2 numbering scheme](https://www.hiv.lanl.gov/content/sequence/HIV/REVIEWS/HXB2.html) for all subsequent steps.\n",
    "\n",
    "In addition, we need to align the Env homologs, and take into account that we only mutagenized a portion of Env (we excluded the N-terminal signal peptide and the C-terminal cytoplasmic tail).\n",
    "\n",
    "First, we define the files holding the Env coding sequences (these are [./data/BG505_env.fasta](./data/BG505_env.fasta) and [./data/BF520_env.fasta](./data/BF520_env.fasta)) as well as the regions of these sequences that were mutagenized in sequential 1, 2, ... numbering."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# the homologs that we are examining\n",
    "homologs = ['BG505', 'BF520']\n",
    "\n",
    "# files with codon sequences\n",
    "wtseqfiles = dict([(env, './data/{0}_env.fasta'.format(env)) for env in homologs])\n",
    "\n",
    "# the mutagenized codon sites in 1, 2, ... numbering\n",
    "# for both this is 31-702 in HXB2 numbering\n",
    "mutagenizedsites = {\n",
    "        'BG505':list(range(30, 699 + 1)),\n",
    "        'BF520':list(range(30, 691 + 1))\n",
    "        }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we read in the aligned sequences of the Env homologs and HXB2.\n",
    "This alignment was created with [mafft](https://mafft.cbrc.jp/alignment/software/) and then manually edited by Hugh Haddox in some of the variable loop regions that align very poorly.\n",
    "The manually edited alignment is in [./data/Env_protalignment_manualtweaks.fasta](./data/Env_protalignment_manualtweaks.fasta).\n",
    "\n",
    "After reading in the aligned proteins from file, we compute the pairwise identity of the two sequences over all non-gapped **mutagenized** sites (we do not include the sites not mutagenized in this calculation):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The BG505 and BF520 proteins are 0.862 identical (732 of 849 alignable sites)\n"
     ]
    }
   ],
   "source": [
    "protalignmentfile = './data/Env_protalignment_manualtweaks.fasta'\n",
    "alignedprots = dict([(seq.name, str(seq.seq)) for seq in \n",
    "        Bio.SeqIO.parse(protalignmentfile, 'fasta')])\n",
    "\n",
    "for (env1, env2) in itertools.combinations(homologs, 2):\n",
    "    (s1, s2) = (alignedprots[env1], alignedprots[env2])\n",
    "    sites = [r for r in range(len(s1)) if s1[r] != '-' and s2[r] != '-']\n",
    "    nident = sum([s1[r] == s2[r] for r in sites])\n",
    "    print(\"The {0} and {1} proteins are {2:.3f} identical ({3} of {4} alignable sites)\"\n",
    "            .format(env1, env2, nident / float(len(sites)), nident, len(sites)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we create files that map each residue in our Env homologs to comparable residue sequential numbering of HXB2.\n",
    "This creates a mapping to the [HXB2 numbering scheme](https://www.hiv.lanl.gov/content/sequence/HIV/REVIEWS/HXB2.html) that we will use later.\n",
    "The renumbering files have columns named *original* (sequential 1, 2, ... numbering) and *new* (HXB2 numbering)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing HXB2 renumbering file for BG505 to ./results/HXB2_numbering/BG505_to_HXB2.csv\n",
      "Writing HXB2 renumbering file for BF520 to ./results/HXB2_numbering/BF520_to_HXB2.csv\n"
     ]
    }
   ],
   "source": [
    "renumbdir = os.path.join(resultsdir, 'HXB2_numbering')\n",
    "if not os.path.isdir(renumbdir):\n",
    "    os.mkdir(renumbdir)\n",
    "\n",
    "for env in homologs:\n",
    "    \n",
    "    # first build numbering with HXB2 numbers potentially repeated\n",
    "    hxb2_r = 1\n",
    "    hxb2_nums = [] # holds HXB2 numbers\n",
    "    for (hxb2_aa, aa) in zip(alignedprots['HXB2'], alignedprots[env]):\n",
    "        if aa != '-':\n",
    "            hxb2_nums.append(str(hxb2_r - int(hxb2_aa == '-')))\n",
    "        if hxb2_aa != '-':\n",
    "            hxb2_r += 1\n",
    "            \n",
    "    # now take all runs of HXB2 numbers and make them 10, 10a, 10b, etc...\n",
    "    for num in set(hxb2_nums):\n",
    "        n_num = hxb2_nums.count(num)\n",
    "        firstnum = hxb2_nums.index(num)\n",
    "        for i in range(n_num - 1):\n",
    "            hxb2_nums[firstnum + i + 1] = num + string.ascii_lowercase[i]\n",
    "                \n",
    "    # now write renumbering file\n",
    "    renumbfile = os.path.join(renumbdir, '{0}_to_HXB2.csv'.format(env))\n",
    "    print(\"Writing HXB2 renumbering file for {0} to {1}\".format(env, renumbfile))\n",
    "    with open(renumbfile, 'w') as f:\n",
    "        f.write('original,new\\n')\n",
    "        for (r, new) in enumerate(hxb2_nums):\n",
    "            f.write('{0},{1}\\n'.format(r + 1, new))      "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download deep sequencing data from the Sequence Read Archive\n",
    "The deep sequencing reads have been submitted to the [Sequence Read Archive (SRA)](http://www.ncbi.nlm.nih.gov/sra), so we download them from there:\n",
    "* The BF520 reads have the BioSample Accession number [SAMN06313000](https://www.ncbi.nlm.nih.gov/sra?term=SAMN06313000) and BioProject number [PRJNA371844](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA371844/).\n",
    "* The BG505 reads have the BioSample Accession number [SAMN07718028](https://www.ncbi.nlm.nih.gov/sra?term=SAMN07718028) and BioProject number [PRJNA412552](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA412552/).\n",
    "\n",
    "We download these files using the [dms_tools2.sra.fastqFromSRA](https://jbloomlab.github.io/dms_tools2/dms_tools2.sra.html#dms_tools2.sra.fastqFromSRA) function from the [dms_tools2 Python API](https://jbloomlab.github.io/dms_tools2/api.html). Note that the call to this function below uses two external programs that you must ensure are installed on the computer that you are using:\n",
    "\n",
    "1. The [fastq-dump](https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&f=fastq-dump) program from the [SRA Toolkit](https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc). If you do not already have this toolkit installed, you will need to install it by following [these instructions](https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&f=std). You need a relatively recent version.\n",
    "\n",
    "2. The [Aspera Connect program](https://www.ncbi.nlm.nih.gov/books/NBK242625/) for rapid downloads. You need both the executable `ascp` and an Aspera Key file. Installing Aspera Connect and a key can be somewhat complex, so if you do not want to do this then just set `aspera=None` in the command below and `fastq-dump` will do the downloads (albeit more slowly).\n",
    "\n",
    "For each homolog, there are the following samples:\n",
    "* the wildtype plasmid controls (*DNA*)\n",
    "* the mutant plasmid DNA libraries (*mutDNA*)\n",
    "* the wildtype virus controls (*virus*)\n",
    "* the mutant virus samples (*mutvirus*)\n",
    "\n",
    "There are three replicates of all of these samples for each homolog, except that the BF520 homolog only has one *DNA* control."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading FASTQ files from the SRA...\n",
      "Here are the names of the downloaded files now found in ./results/FASTQ_files/\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>homolog</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>run</th>\n",
       "      <th>name</th>\n",
       "      <th>R1</th>\n",
       "      <th>R2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>DNA</td>\n",
       "      <td>SRR5241717</td>\n",
       "      <td>BF520-DNA</td>\n",
       "      <td>BF520-DNA_R1.fastq.gz</td>\n",
       "      <td>BF520-DNA_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>mutDNA-1</td>\n",
       "      <td>SRR5241726</td>\n",
       "      <td>BF520-mutDNA-1</td>\n",
       "      <td>BF520-mutDNA-1_R1.fastq.gz</td>\n",
       "      <td>BF520-mutDNA-1_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>mutDNA-2</td>\n",
       "      <td>SRR5241725</td>\n",
       "      <td>BF520-mutDNA-2</td>\n",
       "      <td>BF520-mutDNA-2_R1.fastq.gz</td>\n",
       "      <td>BF520-mutDNA-2_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>mutDNA-3</td>\n",
       "      <td>SRR5241724</td>\n",
       "      <td>BF520-mutDNA-3</td>\n",
       "      <td>BF520-mutDNA-3_R1.fastq.gz</td>\n",
       "      <td>BF520-mutDNA-3_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>virus-1</td>\n",
       "      <td>SRR5241716</td>\n",
       "      <td>BF520-virus-1</td>\n",
       "      <td>BF520-virus-1_R1.fastq.gz</td>\n",
       "      <td>BF520-virus-1_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>virus-2</td>\n",
       "      <td>SRR5241715</td>\n",
       "      <td>BF520-virus-2</td>\n",
       "      <td>BF520-virus-2_R1.fastq.gz</td>\n",
       "      <td>BF520-virus-2_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>virus-3</td>\n",
       "      <td>SRR5241714</td>\n",
       "      <td>BF520-virus-3</td>\n",
       "      <td>BF520-virus-3_R1.fastq.gz</td>\n",
       "      <td>BF520-virus-3_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>mutvirus-1</td>\n",
       "      <td>SRR5241723</td>\n",
       "      <td>BF520-mutvirus-1</td>\n",
       "      <td>BF520-mutvirus-1_R1.fastq.gz</td>\n",
       "      <td>BF520-mutvirus-1_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>mutvirus-2</td>\n",
       "      <td>SRR5241721</td>\n",
       "      <td>BF520-mutvirus-2</td>\n",
       "      <td>BF520-mutvirus-2_R1.fastq.gz</td>\n",
       "      <td>BF520-mutvirus-2_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BF520</td>\n",
       "      <td>mutvirus-3</td>\n",
       "      <td>SRR5241719</td>\n",
       "      <td>BF520-mutvirus-3</td>\n",
       "      <td>BF520-mutvirus-3_R1.fastq.gz</td>\n",
       "      <td>BF520-mutvirus-3_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>DNA-1</td>\n",
       "      <td>SRR6117033</td>\n",
       "      <td>BG505-DNA-1</td>\n",
       "      <td>BG505-DNA-1_R1.fastq.gz</td>\n",
       "      <td>BG505-DNA-1_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>DNA-2</td>\n",
       "      <td>SRR6117029</td>\n",
       "      <td>BG505-DNA-2</td>\n",
       "      <td>BG505-DNA-2_R1.fastq.gz</td>\n",
       "      <td>BG505-DNA-2_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>DNA-3</td>\n",
       "      <td>SRR6117027</td>\n",
       "      <td>BG505-DNA-3</td>\n",
       "      <td>BG505-DNA-3_R1.fastq.gz</td>\n",
       "      <td>BG505-DNA-3_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>mutDNA-1</td>\n",
       "      <td>SRR6117032</td>\n",
       "      <td>BG505-mutDNA-1</td>\n",
       "      <td>BG505-mutDNA-1_R1.fastq.gz</td>\n",
       "      <td>BG505-mutDNA-1_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>mutDNA-2</td>\n",
       "      <td>SRR6117028</td>\n",
       "      <td>BG505-mutDNA-2</td>\n",
       "      <td>BG505-mutDNA-2_R1.fastq.gz</td>\n",
       "      <td>BG505-mutDNA-2_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>mutDNA-3</td>\n",
       "      <td>SRR6117026</td>\n",
       "      <td>BG505-mutDNA-3</td>\n",
       "      <td>BG505-mutDNA-3_R1.fastq.gz</td>\n",
       "      <td>BG505-mutDNA-3_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>virus-1</td>\n",
       "      <td>SRR6117034</td>\n",
       "      <td>BG505-virus-1</td>\n",
       "      <td>BG505-virus-1_R1.fastq.gz</td>\n",
       "      <td>BG505-virus-1_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>virus-2</td>\n",
       "      <td>SRR6117030</td>\n",
       "      <td>BG505-virus-2</td>\n",
       "      <td>BG505-virus-2_R1.fastq.gz</td>\n",
       "      <td>BG505-virus-2_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>virus-3</td>\n",
       "      <td>SRR6117024</td>\n",
       "      <td>BG505-virus-3</td>\n",
       "      <td>BG505-virus-3_R1.fastq.gz</td>\n",
       "      <td>BG505-virus-3_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>mutvirus-1</td>\n",
       "      <td>SRR6117035</td>\n",
       "      <td>BG505-mutvirus-1</td>\n",
       "      <td>BG505-mutvirus-1_R1.fastq.gz</td>\n",
       "      <td>BG505-mutvirus-1_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>mutvirus-2</td>\n",
       "      <td>SRR6117031</td>\n",
       "      <td>BG505-mutvirus-2</td>\n",
       "      <td>BG505-mutvirus-2_R1.fastq.gz</td>\n",
       "      <td>BG505-mutvirus-2_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>BG505</td>\n",
       "      <td>mutvirus-3</td>\n",
       "      <td>SRR6117025</td>\n",
       "      <td>BG505-mutvirus-3</td>\n",
       "      <td>BG505-mutvirus-3_R1.fastq.gz</td>\n",
       "      <td>BG505-mutvirus-3_R2.fastq.gz</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# specify names of all samples to download\n",
    "samples = pandas.DataFrame.from_records(\n",
    "        [('BF520', 'DNA', 'SRR5241717'),\n",
    "         ('BF520', 'mutDNA-1', 'SRR5241726'),\n",
    "         ('BF520', 'mutDNA-2', 'SRR5241725'),\n",
    "         ('BF520', 'mutDNA-3', 'SRR5241724'),\n",
    "         ('BF520', 'virus-1', 'SRR5241716'),\n",
    "         ('BF520', 'virus-2', 'SRR5241715'),\n",
    "         ('BF520', 'virus-3', 'SRR5241714'),\n",
    "         ('BF520', 'mutvirus-1', 'SRR5241723'),\n",
    "         ('BF520', 'mutvirus-2', 'SRR5241721'),\n",
    "         ('BF520', 'mutvirus-3', 'SRR5241719'),\n",
    "         ('BG505', 'DNA-1', 'SRR6117033'),\n",
    "         ('BG505', 'DNA-2', 'SRR6117029'),\n",
    "         ('BG505', 'DNA-3', 'SRR6117027'),\n",
    "         ('BG505', 'mutDNA-1', 'SRR6117032'),\n",
    "         ('BG505', 'mutDNA-2', 'SRR6117028'),\n",
    "         ('BG505', 'mutDNA-3', 'SRR6117026'),\n",
    "         ('BG505', 'virus-1', 'SRR6117034'),\n",
    "         ('BG505', 'virus-2', 'SRR6117030'),\n",
    "         ('BG505', 'virus-3', 'SRR6117024'),\n",
    "         ('BG505', 'mutvirus-1', 'SRR6117035'),\n",
    "         ('BG505', 'mutvirus-2', 'SRR6117031'),\n",
    "         ('BG505', 'mutvirus-3', 'SRR6117025')\n",
    "        ],\n",
    "        columns=['homolog', 'sample_name', 'run']\n",
    "        )\n",
    "samples['name'] = samples['homolog'] + '-' + samples['sample_name']\n",
    "\n",
    "# download to this directory\n",
    "fastqdir = os.path.join(resultsdir, 'FASTQ_files/')\n",
    "\n",
    "print(\"Downloading FASTQ files from the SRA...\")\n",
    "dms_tools2.sra.fastqFromSRA(\n",
    "        samples=samples,\n",
    "        fastq_dump='fastq-dump', # valid path to this program on the Hutch server\n",
    "        fastqdir=fastqdir,\n",
    "        aspera=(\n",
    "            '/app/aspera-connect/3.5.1/bin/ascp', # valid path to ascp on Hutch server\n",
    "            '/app/aspera-connect/3.5.1/etc/asperaweb_id_dsa.openssh' # Aspera key on Hutch server\n",
    "            )\n",
    "        )\n",
    "print(\"Here are the names of the downloaded files now found in {0}\".format(fastqdir))\n",
    "display(HTML(samples.to_html(index=False)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Align the deep sequencing data and count mutations\n",
    "We used [barcoded-subamplicon sequencing](https://jbloomlab.github.io/dms_tools2/bcsubamp.html) to deep sequence the above samples. \n",
    "We use [dms2_batch_bcsubamp](https://jbloomlab.github.io/dms_tools2/dms2_batch_bcsubamp.html#dms2-batch-bcsubamp) to process the FASTQ files to counts of each mutant codon at each site."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Aligning reads and counting mutations for BG505, placing results in ./results/codoncounts/BG505\n",
      "Here is the batch file that we use as input:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>name</th>\n",
       "      <th>R1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>DNA-1</td>\n",
       "      <td>BG505-DNA-1_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA-2</td>\n",
       "      <td>BG505-DNA-2_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA-3</td>\n",
       "      <td>BG505-DNA-3_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutDNA-1</td>\n",
       "      <td>BG505-mutDNA-1_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutDNA-2</td>\n",
       "      <td>BG505-mutDNA-2_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutDNA-3</td>\n",
       "      <td>BG505-mutDNA-3_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>virus-1</td>\n",
       "      <td>BG505-virus-1_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>virus-2</td>\n",
       "      <td>BG505-virus-2_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>virus-3</td>\n",
       "      <td>BG505-virus-3_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutvirus-1</td>\n",
       "      <td>BG505-mutvirus-1_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutvirus-2</td>\n",
       "      <td>BG505-mutvirus-2_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutvirus-3</td>\n",
       "      <td>BG505-mutvirus-3_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now running dms2_batch_bcsubamp...\n",
      "/usr/lib/python3.4/importlib/_bootstrap.py:321: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.\r\n",
      "  return f(*args, **kwds)\r\n",
      "2017-10-25 08:54:33,558 - INFO - Beginning execution of dms2_batch_bcsubamp in directory /fh/fast/bloom_j/computational_notebooks/jbloom/2017/EnvMutationalShiftsPaper/analysis_code\r\n",
      "\r\n",
      "2017-10-25 08:54:33,559 - INFO - Progress is being logged to ./results/codoncounts/BG505/summary.log\r\n",
      "2017-10-25 08:54:35,356 - INFO - Version information:\r\n",
      "\tTime and date: Wed Oct 25 08:54:33 2017\r\n",
      "\tPlatform: Linux-3.13.0-49-generic-x86_64-with-Ubuntu-14.04-trusty\r\n",
      "\tPython version: 3.4.3 (default, Oct 14 2015, 20:28:29)  [GCC 4.8.4]\r\n",
      "\tdms_tools2 version: 2.2.dev0\r\n",
      "\tBio version: 1.70\r\n",
      "\tHTSeq version: 0.9.1\r\n",
      "\tpandas version: 0.20.3\r\n",
      "\tnumpy version: 1.13.1\r\n",
      "\tIPython version: 6.1.0\r\n",
      "\tmatplotlib version: 2.0.2\r\n",
      "\tplotnine version: 0.2.1\r\n",
      "\tnatsort version: 5.1.0\r\n",
      "\tpystan version: 2.16.0.0\r\n",
      "\tscipy version: 0.19.1\r\n",
      "\tseaborn version: 0.8.0\r\n",
      "\tphydmslib version: 2.1.3\r\n",
      "\tjupyter version: 1.0.0\r\n",
      "\r\n",
      "2017-10-25 08:54:35,357 - INFO - Parsed the following arguments:\r\n",
      "\tsitemask = None\r\n",
      "\tR2 = None\r\n",
      "\tbclen = 8\r\n",
      "\tminconcur = 0.75\r\n",
      "\tminreads = 2\r\n",
      "\tchartype = codon\r\n",
      "\tminfraccall = 0.95\r\n",
      "\tfastqdir = ./results/FASTQ_files/\r\n",
      "\tncpus = 6\r\n",
      "\tpurgebc = 0\r\n",
      "\tpurgeread = 0\r\n",
      "\tbcinfo = False\r\n",
      "\toutdir = ./results/codoncounts/BG505\r\n",
      "\tbatchfile = ./results/codoncounts/BG505/batch.csv\r\n",
      "\tR1trim = [200]\r\n",
      "\trefseq = ./data/BG505_env.fasta\r\n",
      "\tuse_existing = yes\r\n",
      "\tminq = 15\r\n",
      "\talignspecs = ['87,375,39,36', '376,666,36,39', '663,954,33,41', '955,1228,33,37', '1228,1527,34,35', '1527,1815,32,39', '1816,2098,36,41']\r\n",
      "\tsummaryprefix = summary\r\n",
      "\tR2trim = [170]\r\n",
      "\tmaxmuts = 4\r\n",
      "\r\n",
      "2017-10-25 08:54:35,565 - INFO - Parsing sample info from ./results/codoncounts/BG505/batch.csv\r\n",
      "2017-10-25 08:54:35,573 - INFO - Read the following sample information:\r\n",
      "name,R1\r\n",
      "DNA-1,BG505-DNA-1_R1.fastq.gz\r\n",
      "DNA-2,BG505-DNA-2_R1.fastq.gz\r\n",
      "DNA-3,BG505-DNA-3_R1.fastq.gz\r\n",
      "mutDNA-1,BG505-mutDNA-1_R1.fastq.gz\r\n",
      "mutDNA-2,BG505-mutDNA-2_R1.fastq.gz\r\n",
      "mutDNA-3,BG505-mutDNA-3_R1.fastq.gz\r\n",
      "virus-1,BG505-virus-1_R1.fastq.gz\r\n",
      "virus-2,BG505-virus-2_R1.fastq.gz\r\n",
      "virus-3,BG505-virus-3_R1.fastq.gz\r\n",
      "mutvirus-1,BG505-mutvirus-1_R1.fastq.gz\r\n",
      "mutvirus-2,BG505-mutvirus-2_R1.fastq.gz\r\n",
      "mutvirus-3,BG505-mutvirus-3_R1.fastq.gz\r\n",
      "\r\n",
      "\r\n",
      "2017-10-25 08:54:35,573 - INFO - Running dms2_bcsubamp on all samples using 6 CPUs...\r\n",
      "2017-10-25 09:14:19,028 - INFO - Completed runs of dms2_bcsubamp.\r\n",
      "\r\n",
      "2017-10-25 09:14:19,051 - INFO - Plotting read stats to ./results/codoncounts/BG505/summary_readstats.pdf\r\n",
      "2017-10-25 09:14:19,981 - INFO - Plotting barcode stats to ./results/codoncounts/BG505/summary_bcstats.pdf\r\n",
      "2017-10-25 09:14:20,591 - INFO - Plotting reads per barcode to ./results/codoncounts/BG505/summary_readsperbc.pdf\r\n",
      "/home/jbloom/.local/lib/python3.4/site-packages/plotnine/ggplot.py:552: UserWarning: Saving 12.92 x 3.12 in image.\r\n",
      "  from_inches(h, units), units))\r\n",
      "2017-10-25 09:14:22,412 - INFO - Plotting count depth to ./results/codoncounts/BG505/summary_depth.pdf\r\n",
      "/home/jbloom/.local/lib/python3.4/site-packages/plotnine/ggplot.py:552: UserWarning: Saving 10.35 x 4.29 in image.\r\n",
      "  from_inches(h, units), units))\r\n",
      "2017-10-25 09:14:24,243 - INFO - Plotting mutation frequencies to ./results/codoncounts/BG505/summary_mutfreq.pdf\r\n",
      "2017-10-25 09:14:35,997 - INFO - Plotting average frequencies of codon mutation types to ./results/codoncounts/BG505/summary_codonmuttypes.pdf, writing the data to ./results/codoncounts/BG505/summary_codonmuttypes.csv\r\n",
      "2017-10-25 09:14:47,061 - INFO - Plotting average frequencies of nucleotide changes per codon mutation to ./results/codoncounts/BG505/summary_codonntchanges.pdf\r\n",
      "2017-10-25 09:14:57,578 - INFO - Plotting frequencies of nucleotide changes in 1-nucleotide mutations to ./results/codoncounts/BG505/summary_singlentchanges.pdf\r\n",
      "2017-10-25 09:15:08,542 - INFO - Plotting fraction of mutations seen less than some number of times to ./results/codoncounts/BG505/summary_cumulmutcounts.pdf\r\n",
      "/home/jbloom/.local/lib/python3.4/site-packages/plotnine/geoms/geom_path.py:76: UserWarning: geom_path: Removed 2770 rows containing missing values.\r\n",
      "  warn(msg.format(n1-n2))\r\n",
      "/home/jbloom/.local/lib/python3.4/site-packages/plotnine/ggplot.py:552: UserWarning: Saving 10.35 x 4.55 in image.\r\n",
      "  from_inches(h, units), units))\r\n",
      "2017-10-25 09:15:27,414 - INFO - Successful completion of dms2_batch_bcsubamp\r\n",
      "Completed dms2_batch_bcsubamp.\n",
      "\n",
      "Aligning reads and counting mutations for BF520, placing results in ./results/codoncounts/BF520\n",
      "Here is the batch file that we use as input:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>name</th>\n",
       "      <th>R1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>DNA</td>\n",
       "      <td>BF520-DNA_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutDNA-1</td>\n",
       "      <td>BF520-mutDNA-1_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutDNA-2</td>\n",
       "      <td>BF520-mutDNA-2_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutDNA-3</td>\n",
       "      <td>BF520-mutDNA-3_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>virus-1</td>\n",
       "      <td>BF520-virus-1_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>virus-2</td>\n",
       "      <td>BF520-virus-2_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>virus-3</td>\n",
       "      <td>BF520-virus-3_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutvirus-1</td>\n",
       "      <td>BF520-mutvirus-1_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutvirus-2</td>\n",
       "      <td>BF520-mutvirus-2_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mutvirus-3</td>\n",
       "      <td>BF520-mutvirus-3_R1.fastq.gz</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now running dms2_batch_bcsubamp...\n",
      "/usr/lib/python3.4/importlib/_bootstrap.py:321: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.\r\n",
      "  return f(*args, **kwds)\r\n",
      "2017-10-25 09:15:33,585 - INFO - Beginning execution of dms2_batch_bcsubamp in directory /fh/fast/bloom_j/computational_notebooks/jbloom/2017/EnvMutationalShiftsPaper/analysis_code\r\n",
      "\r\n",
      "2017-10-25 09:15:33,585 - INFO - Progress is being logged to ./results/codoncounts/BF520/summary.log\r\n",
      "2017-10-25 09:15:34,297 - INFO - Version information:\r\n",
      "\tTime and date: Wed Oct 25 09:15:33 2017\r\n",
      "\tPlatform: Linux-3.13.0-49-generic-x86_64-with-Ubuntu-14.04-trusty\r\n",
      "\tPython version: 3.4.3 (default, Oct 14 2015, 20:28:29)  [GCC 4.8.4]\r\n",
      "\tdms_tools2 version: 2.2.dev0\r\n",
      "\tBio version: 1.70\r\n",
      "\tHTSeq version: 0.9.1\r\n",
      "\tpandas version: 0.20.3\r\n",
      "\tnumpy version: 1.13.1\r\n",
      "\tIPython version: 6.1.0\r\n",
      "\tmatplotlib version: 2.0.2\r\n",
      "\tplotnine version: 0.2.1\r\n",
      "\tnatsort version: 5.1.0\r\n",
      "\tpystan version: 2.16.0.0\r\n",
      "\tscipy version: 0.19.1\r\n",
      "\tseaborn version: 0.8.0\r\n",
      "\tphydmslib version: 2.1.3\r\n",
      "\tjupyter version: 1.0.0\r\n",
      "\r\n",
      "2017-10-25 09:15:34,298 - INFO - Parsed the following arguments:\r\n",
      "\trefseq = ./data/BF520_env.fasta\r\n",
      "\tR1trim = [205]\r\n",
      "\tbclen = 8\r\n",
      "\tminreads = 2\r\n",
      "\toutdir = ./results/codoncounts/BF520\r\n",
      "\tuse_existing = yes\r\n",
      "\tminq = 15\r\n",
      "\tmaxmuts = 4\r\n",
      "\tR2trim = [212]\r\n",
      "\tchartype = codon\r\n",
      "\tsitemask = None\r\n",
      "\tbatchfile = ./results/codoncounts/BF520/batch.csv\r\n",
      "\tpurgebc = 0\r\n",
      "\talignspecs = ['88,417,40,39', '418,753,39,35', '754,1092,36,39', '1093,1447,35,29', '1407,1758,35,32', '1759,2097,30,36']\r\n",
      "\tminfraccall = 0.95\r\n",
      "\tsummaryprefix = summary\r\n",
      "\tminconcur = 0.75\r\n",
      "\tpurgeread = 0\r\n",
      "\tfastqdir = ./results/FASTQ_files/\r\n",
      "\tncpus = 6\r\n",
      "\tbcinfo = False\r\n",
      "\tR2 = None\r\n",
      "\r\n",
      "2017-10-25 09:15:34,322 - INFO - Parsing sample info from ./results/codoncounts/BF520/batch.csv\r\n",
      "2017-10-25 09:15:34,329 - INFO - Read the following sample information:\r\n",
      "name,R1\r\n",
      "DNA,BF520-DNA_R1.fastq.gz\r\n",
      "mutDNA-1,BF520-mutDNA-1_R1.fastq.gz\r\n",
      "mutDNA-2,BF520-mutDNA-2_R1.fastq.gz\r\n",
      "mutDNA-3,BF520-mutDNA-3_R1.fastq.gz\r\n",
      "virus-1,BF520-virus-1_R1.fastq.gz\r\n",
      "virus-2,BF520-virus-2_R1.fastq.gz\r\n",
      "virus-3,BF520-virus-3_R1.fastq.gz\r\n",
      "mutvirus-1,BF520-mutvirus-1_R1.fastq.gz\r\n",
      "mutvirus-2,BF520-mutvirus-2_R1.fastq.gz\r\n",
      "mutvirus-3,BF520-mutvirus-3_R1.fastq.gz\r\n",
      "\r\n",
      "\r\n",
      "2017-10-25 09:15:34,330 - INFO - Running dms2_bcsubamp on all samples using 6 CPUs...\r\n",
      "2017-10-25 09:30:33,724 - INFO - Completed runs of dms2_bcsubamp.\r\n",
      "\r\n",
      "2017-10-25 09:30:33,822 - INFO - Plotting read stats to ./results/codoncounts/BF520/summary_readstats.pdf\r\n",
      "2017-10-25 09:30:34,581 - INFO - Plotting barcode stats to ./results/codoncounts/BF520/summary_bcstats.pdf\r\n",
      "2017-10-25 09:30:35,104 - INFO - Plotting reads per barcode to ./results/codoncounts/BF520/summary_readsperbc.pdf\r\n",
      "/home/jbloom/.local/lib/python3.4/site-packages/plotnine/ggplot.py:552: UserWarning: Saving 12.92 x 3.12 in image.\r\n",
      "  from_inches(h, units), units))\r\n",
      "2017-10-25 09:30:36,842 - INFO - Plotting count depth to ./results/codoncounts/BF520/summary_depth.pdf\r\n",
      "/home/jbloom/.local/lib/python3.4/site-packages/plotnine/ggplot.py:552: UserWarning: Saving 10.35 x 4.29 in image.\r\n",
      "  from_inches(h, units), units))\r\n",
      "2017-10-25 09:30:38,436 - INFO - Plotting mutation frequencies to ./results/codoncounts/BF520/summary_mutfreq.pdf\r\n",
      "2017-10-25 09:30:47,858 - INFO - Plotting average frequencies of codon mutation types to ./results/codoncounts/BF520/summary_codonmuttypes.pdf, writing the data to ./results/codoncounts/BF520/summary_codonmuttypes.csv\r\n",
      "2017-10-25 09:30:56,328 - INFO - Plotting average frequencies of nucleotide changes per codon mutation to ./results/codoncounts/BF520/summary_codonntchanges.pdf\r\n",
      "2017-10-25 09:31:04,658 - INFO - Plotting frequencies of nucleotide changes in 1-nucleotide mutations to ./results/codoncounts/BF520/summary_singlentchanges.pdf\r\n",
      "2017-10-25 09:31:13,097 - INFO - Plotting fraction of mutations seen less than some number of times to ./results/codoncounts/BF520/summary_cumulmutcounts.pdf\r\n",
      "/home/jbloom/.local/lib/python3.4/site-packages/plotnine/geoms/geom_path.py:76: UserWarning: geom_path: Removed 2168 rows containing missing values.\r\n",
      "  warn(msg.format(n1-n2))\r\n",
      "/home/jbloom/.local/lib/python3.4/site-packages/plotnine/ggplot.py:552: UserWarning: Saving 10.35 x 4.55 in image.\r\n",
      "  from_inches(h, units), units))\r\n",
      "2017-10-25 09:31:28,159 - INFO - Successful completion of dms2_batch_bcsubamp\r\n",
      "Completed dms2_batch_bcsubamp.\n"
     ]
    }
   ],
   "source": [
    "# alignment specs for each homolog\n",
    "alignspecs = {\n",
    "        'BG505':' '.join(\n",
    "                   ['87,375,39,36',\n",
    "                    '376,666,36,39',\n",
    "                    '663,954,33,41',\n",
    "                    '955,1228,33,37',\n",
    "                    '1228,1527,34,35',\n",
    "                    '1527,1815,32,39',\n",
    "                    '1816,2098,36,41']),\n",
    "        'BF520':' '.join(\n",
    "                   ['88,417,40,39', \n",
    "                    '418,753,39,35',\n",
    "                    '754,1092,36,39',\n",
    "                    '1093,1447,35,29',\n",
    "                    '1407,1758,35,32',\n",
    "                    '1759,2097,30,36'])\n",
    "        }\n",
    "\n",
    "# read trimming for each homolog\n",
    "r1trim = {'BG505':'200', 'BF520':'205'}\n",
    "r2trim = {'BG505':'170', 'BF520':'212'}\n",
    "\n",
    "# define counts directory containing output for each homolog\n",
    "basecountsdir = os.path.join(resultsdir, 'codoncounts')\n",
    "if not os.path.isdir(basecountsdir):\n",
    "    os.mkdir(basecountsdir)\n",
    "countsdir = {}\n",
    "for env in homologs:\n",
    "    countsdir[env] = os.path.join(basecountsdir, env)\n",
    "    if not os.path.isdir(countsdir[env]):\n",
    "        os.mkdir(countsdir[env])\n",
    "\n",
    "# align each homolog\n",
    "for env in homologs:\n",
    "    print('\\nAligning reads and counting mutations for {0}, '\n",
    "          'placing results in {1}'.format(env, countsdir[env]))\n",
    "    \n",
    "    countsbatchfile = os.path.join(countsdir[env], 'batch.csv')\n",
    "    print(\"Here is the batch file that we use as input:\")\n",
    "    countsbatch = (samples.query('homolog == @env')\n",
    "                          [['sample_name', 'R1']]\n",
    "                          .rename(columns={'sample_name':'name'})\n",
    "                          )\n",
    "    display(HTML(countsbatch.to_html(index=False)))\n",
    "    countsbatch.to_csv(countsbatchfile, index=False)\n",
    "    \n",
    "    # make a sitemask that includes only mutagenized sites\n",
    "    sitemask = os.path.join(countsdir[env], 'sitemask.csv')\n",
    "    with open(sitemask, 'w') as f:\n",
    "        f.write('\\n'.join(map(str, ['site'] + mutagenizedsites[env])))\n",
    "\n",
    "    print('\\nNow running dms2_batch_bcsubamp...')\n",
    "    !dms2_batch_bcsubamp \\\n",
    "            --batchfile {countsbatchfile} \\\n",
    "            --refseq {wtseqfiles[env]} \\\n",
    "            --alignspecs {alignspecs[env]} \\\n",
    "            --outdir {countsdir[env]} \\\n",
    "            --summaryprefix summary \\\n",
    "            --R1trim {r1trim[env]} \\\n",
    "            --R2trim {r2trim[env]} \\\n",
    "            --fastqdir {fastqdir} \\\n",
    "            --ncpus {ncpus} \\\n",
    "            --use_existing {use_existing} \n",
    "    print(\"Completed dms2_batch_bcsubamp.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
