%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ELIFE ARTICLE TEMPLATE
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% PREAMBLE 
\documentclass[9pt]{elife}
\sloppy
% add lineno for line numbering
% Use the onehalfspacing option for 1.5 line spacing
% Use the doublespacing option for 2.0 line spacing
% Please note that these options may affect formatting.


\usepackage{lipsum} % Required to insert dummy text
\usepackage[version=4]{mhchem}
\usepackage{siunitx}
\DeclareSIUnit\Molar{M}
%\usepackage{indentfirst} %I prefer to have indents following section headings, I find it to be more consistent in style throughout document
\usepackage{natbib}
\usepackage{float}
\usepackage{tikz}
\usepackage{seqsplit}

\usepackage{amsmath} % or simply amstext
\newcommand{\angstrom}{\textup{\AA}}


\newfloat{suppfile}{thp}{lofsupfile}
%\renewcommand{\thesuppfile}{Supplementary file \arabic{suppfile}}
\floatname{suppfile}{Supplementary file}
\usepackage{placeins}

\renewcommand{\topfraction}{1}
\renewcommand{\bottomfraction}{1}
\renewcommand{\textfraction}{0}
\renewcommand{\floatpagefraction}{1}
\setcounter{topnumber}{9}
\setcounter{bottomnumber}{9}
\setcounter{totalnumber}{20}

\newcommand{\jdbcomment}[1]{\emph{\color{red} [#1]}}
\newcommand{\hhcomment}[1]{\emph{\color{blue} [#1]}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ARTICLE SETUP
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{Mapping mutational effects along the evolutionary landscape of HIV envelope}

\author[1,2,\authfn{1}]{Hugh K. Haddox}
\author[1,2,\authfn{1}]{Adam S. Dingens}
\author[1,3]{Sarah K. Hilton}
\author[4]{Julie Overbaugh}
\author[1,3,*]{Jesse D. Bloom}
\affil[1]{Basic Sciences Division and Computational Biology Program, Fred Hutchinson Cancer Research Center, Seattle, WA} 
\affil[2]{Molecular and Cellular Biology PhD program, University of Washington, Seattle, WA}
\affil[3]{Department of Genome Sciences, University of Washington, Seattle, WA}
\affil[4]{Human Biology Division and Epidemiology Program, Fred Hutchinson Cancer Research Center, Seattle, WA}
\contrib[\authfn{1}]{These authors contributed equally to this work}

\corr{jbloom@fredhutch.org}{JDB}
% \presentadd[\authfn{5}]{eLife Sciences editorial Office, eLife Sciences, Cambridge, United Kingdom}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ARTICLE START
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\maketitle

\begin{abstract}
The immediate evolutionary space accessible to HIV is largely determined by how single amino-acid mutations affect fitness.
These mutational effects can shift as the virus evolves.
However, the prevalence of such shifts in mutational effects remains unclear.
Here we quantify the effects on viral growth of all amino-acid mutations to two HIV envelope (Env) proteins that differ at $>$100 residues.
Most mutations similarly affect both Envs, but the amino-acid preferences of a minority of sites have clearly shifted.
These shifted sites usually prefer a specific amino acid in one Env, but tolerate many amino acids in the other.
Surprisingly, shifts are only slightly enriched at sites that have substituted between the Envs -- and many occur at residues that do not even contact substitutions.
Therefore, long-range epistasis can unpredictably shift Env's mutational tolerance during HIV evolution, although the amino-acid preferences of most sites are conserved between moderately diverged viral strains.
\end{abstract}


\section{Introduction}
HIV's envelope (Env) protein evolves very rapidly.
The major group of HIV-1 that is responsible for the current pandemic originated from a virus that entered the human population $\sim$100 years ago~\citep{sharp2011origins,worobey2008direct,faria2014early}.
The descendants of this virus have evolved so rapidly that their Envs now have as little as 65\% protein identity~\citep{lynch2009appreciating}.
For comparison, protein orthologs shared between humans and mice have only diverged to a median identity of 78\% over 90 million years~\citep{waterston2002initial,hedges2006timetree}.

Env's rapid evolution has dire consequences for anti-HIV immunity, since it erodes the efficacy of most neutralizing antibodies~\citep{albert1990rapid,wei2003antibody,richman2003rapid,burton2005antibody}.
Because of this public-health importance, numerous studies have experimentally characterized aspects of the ``evolutionary landscape'' that Env traverses.
The immediate evolutionary space accessible to any given Env is largely defined by the effects on viral fitness of all single amino-acid mutations to Env.
Most mutational studies have measured how just a small number of these mutations affect viral growth in cell culture, although it has recently become possible to use deep mutational scanning to measure the effects of many~\citep{al2014high,duenas2016saturation} or even all~\citep{haddox2016experimental} single amino-acid mutations mutations to an Env variant.

But interpreting these studies in the context of Env evolution requires addressing a fundamental question: How informative are mutational studies of a single protein variant about constraints on long-term evolution?
During protein evolution, substitutions at one site can change the effect of mutations at other sites~\citep{natarajan2013epistasis,gong2013stability,harms2014historical,podgornaia2015pervasive,starr2016epistasis,klink2017parallel}.
We will follow the nomenclature of \citet{pollock2012amino} to refer to these changes in mutational effects as \emph{shifts} in a site's amino-acid preferences.
Such shifts can accumulate as substitutions become entrenched via epistatic interactions with subsequent changes~\citep{starr2017pervasive,pollock2012amino,shah2015contingency,bazykin2015changing} -- although the magnitude of these shifts is usually limited~\citep{doud2015site,chan2017correlation,ashenberg2013mutational,risso2014mutational}.

Given that the Envs of circulating HIV strains represent a vast collection of homologs that often differ at $>$100 residues, shifts in amino-acid preferences could make the outcome of any study highly dependent on the Env used.
The extent to which this is actually the case is unclear, since the few protein-wide studies of such shifts have examined proteins that are structurally far simpler than Env, which forms a large heavily glycosylated heterotrimeric complex that transitions through multiple conformational states~\citep{munro2014conformational,ozorowski2017open}.

Here we use an improved version of a previously described deep mutational scanning strategy~\citep{haddox2016experimental} to measure the effects on viral growth of all single amino-acid mutations to two transmitted-founder virus Envs that differ by $>$100 mutations.
We compare these complete maps of mutational effects to identify sites that have shifted in their amino-acid preferences between the Envs.
Most sites show no detectable shifts, but 30 sites have clearly shifted preferences.
These shifted sites usually prefer a specific amino acid in one Env but have shifted to tolerate many amino acids in the other Env.
The shifted sites cluster in structure, but are often distant from any amino-acid substitutions that distinguish the two Envs, demonstrating the action of long-range epistasis.
By aggregating our measurements for both Envs, we identify sites that evolve faster or slower in nature than expected given the functional constraints measured in the lab, probably due to pressure for immune evasion.
Overall, our work provides complete across-strain maps of mutational effects that inform analyses of Env's evolution and function.

\section{Results}

\subsection{Two Envs from clade A transmitted-founder viruses}
The viruses most relevant to HIV's long-term evolution are those which are transmitted from human-to-human.
However, the only prior work that has measured how all Env amino-acid mutations affect HIV growth is a study by some of us~\citep{haddox2016experimental} that used a late-stage lab-passaged CXCR4-tropic virus~\citep[LAI;][]{peden1991changes}.
The properties of Env can vary substantially between such late-stage viruses and the transmitted-founder viruses relevant to HIV's long-term evolution~\citep{sagar2006human,wilen2011phenotypic,parrish2013phenotypic,ronen2015hiv}.

\begin{figure}
\centerline{\includegraphics[clip=true, trim= 1in 0in 1.4in 0in, angle=-90, width=0.85\textwidth]{figures/tree_plot.pdf}}
\caption{\label{fig:tree}
Phylogenetic tree showing the relationship of BG505 and BF520 to other clade A Envs.
The tree shows the 69 Envs in the alignment in \FIG{tree}-source~data~\ref{figdata:alignment}, which is a subsample of clade A sequences from the group M alignment in the Los Alamos HIV sequence database (\url{http://www.hiv.lanl.gov}).
Sites not mutagenized in our experiments (the signal peptide and cytoplasmic tail) or that are poorly alignable were masked as indicated in \FIG{tree}-source~data~\ref{figdata:mask}, leaving 616 alignable sites.
The pairwise identity of BG505 and BF520 to other sequences at alignable sites is in \FIGSUPP[tree]{identity}.
The tree topology was inferred using \texttt{RAxML}~\citep{stamatakis2014raxml} under the GTRCAT model of nucleotide substitution, and branch lengths were optimized under the M0 Goldman-Yang model~\citep{yang2000codon} using \texttt{phydms}~\citep{hilton2017phydms}.
}
\figsupp[\label{figsupp:identity}  Pairwise identity of all Env sequences to BG505 and BF520.]
{The histograms show the pairwise amino-acid identity of each Env to all other sequences in the clade A alignment in \FIG{tree}-source~data~\ref{figdata:alignment} after masking the sites delineated in \FIG{tree}-source~data~\ref{figdata:mask}.
There are 616 non-masked sites.
The pairwise protein identity between BG505 and BF520 is 86.2\% (721 of 836 sites identical) when considering \emph{all} sites, and 89.1\% (549 of 616 sites identical) when considering just the non-masked sites.}
{\includegraphics[width=0.7\textwidth]{figures/masked_alignment_identity.pdf}}
\figdata{\label{figdata:alignment} The alignment of clade A \textit{env} coding sequences is in \texttt{cladeA\_alignment.fasta}.}
\figdata{\label{figdata:mask} The 240 Env sites masked in all phylogenetic analyses because they were not mutagenized in our experiments or are poorly alignable are listed in \texttt{alignment\_mask.csv}.}
\end{figure}

We therefore selected Envs from two transmitted-founder viruses, BG505.W6M.C2.T332N and BF520.W14M.C2 (hereafter referred to as BG505 and BF520), that were isolated from HIV-infected infants shortly after mother-to-child transmission~\citep{nduati2000effect,wu2006neutralization,goo2014early}.
The BG505 Env has been extensively studied from a structural standpoint~\citep{julien2013crystal,lyumkis2013cryo,pancera2014structure,huang2014broad,sanders2015hiv,stewart2016trimeric,gristick2016natively}, and variants of this Env are being tested as a vaccine immunogens~\citep{sanders2013next,sanders2015hiv,de2015immunogenicity}.
We used the T332N variant of BG505 Env because it has a common glycosylation site that is targeted by many anti-HIV antibodies~\citep{sanders2013next}.
The BF520 Env was isolated from an infant who developed an early broad anti-HIV antibody response~\citep{goo2014early,simonich2016hiv}.
We have previously created comprehensive codon-mutant libraries of the BF520 Env and used them to map HIV antibody escape~\citep{dingens2017comprehensive}, but these BF520 libraries have not been characterized with respect to how mutations affect viral growth.

Both BG505 and BF520 are from clade A of the major (M) group of HIV-1.
\FIG{tree} shows the phylogenetic relationship among these two Envs and other clade A sequences.
BG505 and BF520 are identical at 721 of the 836 pairwise-alignable protein sites (86.2\% identity).
In our experiments, we analyze only the ectodomain and transmembrane domain of Env (we exclude the signal peptide and cytoplasmic tail).
In these regions of Env, BG505 and BF520 are identical at 549 of the 616 sites (89.1\% identity) that are alignable across clade A Envs (\FIG{tree}-source~data~\ref{figdata:alignment}, \FIG{tree}-source~data~\ref{figdata:mask}).
The divergence between BG505 and BF520 therefore offers ample opportunity to investigate mutational shifts during Env evolution.

\subsection{Deep mutational scanning of each Env}
We have previously described a deep mutational scanning strategy for measuring how all amino-acid mutations to Env affect HIV growth in cell culture, and applied this strategy to the late-stage lab-adapted LAI strain~\citep{haddox2016experimental}.
Here we made several modifications to this earlier strategy to apply it to transmitted-founder Envs and to reduce the experimental noise. 
This last consideration is especially important when comparing Envs, since it is only possible to reliably detect differences that exceed the magnitude of the experimental noise. 
Our modified deep mutational scanning strategy is in \FIG{dms_schematic}A.
This approach had the following substantive changes: we used SupT1 cells expressing CCR5 to support growth of viruses with transmitted-founder Envs, we used more virions for the first passage ($\ge 3 \times 10^6$ versus $5\times 10^5$ infectious units per library) to avoid bottlenecking library diversity, and rather than performing a full second passage we just did a short high-MOI infection to enable recovery of \textit{env} genes from infectious virions without bottlenecking (\FIG{dms_schematic}A).
We performed this deep mutational scanning in full biological triplicate for both BG505 and BF520 (\FIG{dms_schematic}B).
Our libraries encompassed all codon mutations to all sites in Env except for the signal peptide and cytoplasmic tail.

\begin{figure}
\centerline{\includegraphics[width=0.75\textwidth]{figures/dms_schematic/dms_schematic.pdf}}
\caption{\label{fig:dms_schematic}
Deep mutational scanning workflow.
{\bf (A)} We made libraries of proviral HIV plasmids with random codon-level mutations in the \textit{env} gene.
The number of mutations per gene approximately followed a Poisson distribution with a mean between 1 and 1.5 (\FIGSUPP[dms_schematic]{fsanger}).
We transfected the plasmids into 293T cells to generate mutant viruses, which lack a genotype-phenotype link since cells are multiply transfected.
To establish a genotype-phenotype link and select for Env variants that support HIV growth, we passaged the libraries in SupT1.CCR5 cells for four days at a low multiplicity of infection (MOI) of 0.01.
To isolate the \textit{env} genes from only viruses that encoded a functional Env protein, we infected the passaged libraries into SupT1.CCR5 cells at high MOI and harvested reverse-transcribed non-integrated viral DNA after 12 hours.
We then deep sequenced the \textit{env} genes from these final samples as well as the initial plasmid library, using molecular barcoding to reduce sequencing errors.
We also deep sequenced identically handled wildtype controls to estimate error rates.
Using these sequencing data, we estimated the preference for each of the 20 amino acids at each site in Env.
These data are represented in logo plots, with the height of each letter proportional to that site's preference for that amino acid.
{\bf (B)} We conducted this experiment in full biological triplicate for both BG505 and BF520, beginning each replicate with independent creation of the plasmid mutant library.
These replicates therefore account for \emph{all} sources of noise and error in the experiments.
}
\figsupp[Sanger sequencing of selected clones from the mutant plasmid libraries.\label{figsupp:sanger}]{
We Sanger sequenced 44 clones of BG505 Env sampled roughly evenly from each of the three replicate mutant plasmid libraries.
{\bf(A)} There was an average of 1.5 mutant codons per clone, with the number of mutations per clone roughly following a Poisson distribution. 
{\bf(B)} The mutant codons had a mix of single-, double-, and triple-nucleotide changes, with an elevated number of single-nucleotide changes than expected.
{\bf(C)} Nucleotide frequencies were fairly uniform in the mutant codons.
{\bf(D)} Mutations were distributed roughly evenly along the mutagenized region of {\it env} (30-699 in the sequential numbering scheme used in this plot).
{\bf(E)} For clones with multiple mutations, we computed the pairwise distance in primary sequence between each codon mutation and plotted the cumulative distribution of these distances (red line). 
We also simulated the expected distribution of pairwise distances if mutations occurred entirely independently (blue line). 
The observed distribution is close to the expected distribution.
Comparable data for the BF520 libraries is provided in \citet{dingens2017comprehensive}.
}{\includegraphics[width=\textwidth]{figures/sanger_sequencing_supp/sanger_sequencing_supp.pdf}}
\end{figure}

\begin{figure}
\begin{minipage}[t]{0.34\textwidth}
{\bf \Large A} \\ 
\centerline{{\includegraphics[clip=true, trim=4.3in 1.4in 0in 0.48in, width=0.4\textwidth]{figures/BG505_avgmutfreqs.pdf}}} 
\centerline{\small BG505}
\includegraphics[clip=true, trim=0in 0in 1.7in 0in, width=\textwidth]{figures/BG505_avgmutfreqs.pdf}
\centerline{\small BF520}
\includegraphics[clip=true, trim=0in 0in 1.7in 0in, width=0.89\textwidth]{figures/BF520_avgmutfreqs.pdf}
\end{minipage}
\begin{minipage}[t]{0.01\textwidth}
\end{minipage}
\begin{minipage}[t]{0.65\textwidth}
{\bf \Large B} \\
\includegraphics[width=\textwidth]{figures/allprefscorr.pdf}
\end{minipage}
\caption{\label{fig:mutfreqs}
The deep mutational scanning selects for functional Envs and yields measurements that are well correlated among replicates.
{\bf (A)}
The average per-codon mutation frequency when sequencing plasmids encoding wildtype Env (\emph{DNA}), plasmid mutant libraries (\emph{mutDNA}), mutant viruses after the final infection (\emph{mutvirus}), and virus generated from wildtype plasmids (\emph{virus}).
Mutations are categorized as nonsynonymous, synonymous, or stop codon. 
The \emph{DNA} samples show that sequencing errors are rare, and the \emph{virus} samples show that viral-replication errors are well below the frequency of mutations in the \emph{mutDNA} samples.
Comparing the \emph{mutvirus} to \emph{mutDNA} shows clear purifying selection against stop codons and some nonsynonymous mutations, particularly after subtracting the background error rates given by the \emph{virus} and \emph{DNA} samples (\FIGSUPP[mutfreqs]{mutfreqs}).
More extensive plots from the analysis of the deep sequencing data are in Supplementary~files~\ref{suppfile:code} and \ref{suppfile:html}.
{\bf (B)}
Correlations between replicates in the measured preferences of each site in Env for all 20 amino acids.
Blue indicates replicate measurements on BF520, red indicates replicate measurements on BG505, and gray indicates across-Env measurements of BF520 versus BG505.
$R$ is the Pearson correlation coefficient. 
The numerical values for the preferences are in \FIG{mutfreqs}-source~data~\ref{figdata:allprefs}.
\FIGSUPP[mutfreqs]{contour} shows the correlations using contour rather than scatter plots.
}
\figsupp[\label{figsupp:mutfreqs} Numerical frequencies of mutations.]
{This table gives the average frequencies of nonsynonymous, synonymous, and stop-codon mutations as plotted in \FIG{mutfreqs}.
Note that there is only one \emph{DNA} sample for BF520, so that same sample is repeated three times in the sample in conjunction with each BF520 replicate.
We calculate the error-corrected \emph{pre}-selection mutation frequency as the \emph{mutDNA} frequency minus the \emph{DNA} frequency, and the error-corrected \emph{post}-selection mutation frequency as the \emph{mutvirus} frequency minus the \emph{virus} frequency.
We then use these error-corrected frequencies to calculate the \emph{percent} of mutations remaining after selection for each replicate and type of mutation.}
{\input{figures/avgmutfreqs}}
\figsupp[\label{figsupp:contour} Correlations plotted on a contour rather than a scatter plot.]
{
This figure shows the same data as in \FIG{mutfreqs}B except that it shows KDE contour plots rather than points.
As is obvious from this representation, the vast majority of the points fall near the origin (very low preference in both replicates), and most of the correlation signal is therefore due to the relatively modest number of amino acids that have high preference at any given site.
This is expected, since most amino acid mutations at most sites will be strongly deleterious, and so have low preference.
}{\includegraphics[width=\textwidth]{figures/allprefscorr_contour.pdf}}
\figdata{\label{figdata:allprefs} Preferences for each replicate and averages are in \texttt{all\_prefs\_unscaled.zip}.}
\end{figure}

The deep mutational scanning effectively selected for functional Envs as evidenced by strong purifying selection against stop codons.
\FIG{mutfreqs}A shows the average frequency of mutations across Env in the plasmid mutant libraries, the mutant viruses, and wildtype controls as determined from the deep sequencing.
The mutant viruses show clear selection against stop codons and many nonsynonymous mutations (\FIG{mutfreqs}A).
This selection is more apparent if we correct for the background error rates estimated from the wildtype controls (\FIGSUPP[mutfreqs]{mutfreqs}).
The error-corrected frequencies of stop codons drop to 3\%--16\% of their original values (\FIGSUPP[mutfreqs]{mutfreqs}), with the residual stop codons probably due to some non-functional virions surviving due to complementation by other co-infecting virions. 
The error-corrected frequencies of nonsynonymous mutations also drop substantially (43\%--49\% of their original values), whereas the frequencies of synonymous mutations drop only slightly (85\%--95\% of their original values).
These trends are consistent with the fact that nonsynonymous mutations are often deleterious, whereas synonymous mutations often~\citep[although certainly not always, see][]{zanini2013quantifying} have only mild effects on viral growth.
\FIG{mutfreqs}A only summarizes one aspect of the deep mutational scanning data, but Supplementary~files~\ref{suppfile:code} and \ref{suppfile:html} contain detailed plots showing all aspects of the data (read depth, per-site mutation rate, etc) as generated by the \texttt{dms\_tools2} software~\citep[\url{https://jbloomlab.github.io/dms_tools2/}]{bloom2015software}.

We used the deep mutational scanning data to estimate the preference of each site in Env for each amino acid via the analysis method described in \citet{bloom2015software}.
As graphically illustrated in \FIG{dms_schematic}A, the preferences for each site are normalized to sum to one.
Our libraries were mutagenized at 670 sites in BG505 and 662 sites in BF520, so $670 \times 20 = 13,400$ and $662 \times 20 = 13,240$ preferences were estimated for each Env, respectively.
The correlations between the preferences from different experimental replicates are in \FIG{mutfreqs}B.
These replicate-to-replicate correlations are substantially higher than those for the deep mutational scanning of LAI Env by \citet{haddox2016experimental}, which had replicate-to-replicate Pearson correlations of only $R = 0.45$ to $0.50$. 

While the replicates are well correlated across all replicates for both BG505 and BF520, the replicates for BG505 are more correlated with each other than with replicates for BF520, and vice versa (\FIG{mutfreqs}B, compare red and blue versus gray plots).
This fact hints that there are some shifts in amino-acid preferences between the two Envs -- something that is investigated with more statistical rigor later in this paper.
Note also that there is a trend for highly preferred amino acids to be more strongly preferred in BG505 than BF520 (most high-preference points in the gray plots in \FIG{mutfreqs}B fall above the diagonal); however, this trend does not necessarily reflect differences between the Envs.
Rather, there were modest differences in the stringency of selection between our deep mutational scans of BG505 and BF520 (\FIGSUPP[mutfreqs]{mutfreqs} shows that purifying selection better purged stop codons in BG505).
In the next section, we correct for these experimental differences by calibrating each dataset to match the stringency of selection in nature.

\subsection{Amino-acid preferences of the Envs and their relationship to HIV evolution}
The most immediate question is how authentically the experimental measurements describe the actual selection on Env function in nature.
Direct comparisons between experimentally measured amino-acid preferences and amino-acid frequencies in natural sequences are confounded by the fact that the natural sequences are evolutionarily related.
This problem can be overcome by making the comparison in a phylogenetic context.
Specifically, we used our deep mutational scanning data to construct experimentally informed codon models (ExpCM's) for Env evolution~\citep{hilton2017phydms}.
Importantly, since we expect many sites in Env to be under diversifying selection from immunity, we extended the ExpCM's to draw the relative dN/dS parameter ($\omega$) from a gamma distribution as is commonly done for codon-substitution models~\citep{yang2000codon}.

\begin{table}
\begin{fullwidth}
{\centering \include{figures/gamma_omega_modelcomparison}}
\caption{\label{tab:phydms}
Evolutionary models informed by the deep mutational scanning describe HIV's evolution in nature much better than a standard substitution model.
Shown are the results of maximum likelihood fitting of substitution models to the clade A phylogeny in \FIG{tree}.
Experimentally informed codon models~\citep[ExpCM,][]{hilton2017phydms} utilizing the across-replicate average of the deep mutational scanning describe Env's natural evolution far better than a standard codon substitution model~\citep[the M5 model of][]{yang2000codon} as judged by comparing the Akaike information criteria~\citep[$\Delta$AIC,][]{posada2004model}.
Both ExpCM models have a stringency parameter $>$1.
All models draw $\omega$ from a gamma distribution, and the table shows the mean ($\overline{\omega}$) and shape parameters ($\omega_{\alpha}$ and $\omega_{\beta}$) of this distribution.
The last two columns show the number of sites evolving faster ($\omega_r > 1$) or slower ($\omega_r < 1$) than expected at a false discovery rate of 0.05, as determined using the approach in \citet{bloom2017identification}.
Analyses were performed using \texttt{phydms} (\url{http://jbloomlab.github.io/phydms/}).
\TABLE{phydms}-source-data~\ref{tabledata:phydms} shows the results for additional substitution models.
}
\tabledata{\label{tabledata:phydms}Results for phylogenetic models where $\omega$ is not drawn from a gamma-distribution or where the preferences are averaged across sites to eliminate the site specificity are in \texttt{modelcomparison.md}.}
\end{fullwidth}
\end{table}

\TABLE{phydms} shows that phylogenetic models informed by the deep mutational scanning of either BG505 or BF520 describe the natural evolution of Env vastly better than a standard codon substitution model.
In addition to simply comparing model fits, we can interpret the model parameters.
ExpCM's have a stringency parameter that relates selection in the experiments to that in nature.
A stringency parameter $>$1 indicates that natural selection prefers the same amino acids as the experiments, but with greater stringency~\citep{hilton2017phydms}.
Both ExpCM's have stringency parameters $>$1 (\TABLE{phydms}) -- a finding that makes sense, since the stop-codon analysis in the previous section suggests that the experimental selections are more lax than natural selection on HIV.
We can also interpret the $\omega$ parameter.
For standard codon substitution models, $\omega$ is the rate of fixation of nonsynonymous mutations relative to synonymous ones -- and the gene-wide average $\omega$ is almost always $<$1, since purifying selection purges many functionally deleterious amino-acid mutations even for adaptively evolving proteins~\citep{murrell2015gene}.
Indeed, \TABLE{phydms} shows that Env's gene-wide average $\omega$ is $<$1 for a standard model.
But for ExpCM's, $\omega$ is the relative rate of nonsynonymous to synonymous substitutions \emph{after} accounting for functional constraints measured in the deep mutational scanning~\citep{bloom2017identification}.
\TABLE{phydms} shows that the ExpCM's have a gene-wide average $\omega$ that is $>$1, indicating that external selection (e.g., from immunity) drives Env to fix amino-acid mutations faster than expected under a null model that only accounts for functional constraints.

\begin{figure}
\begin{fullwidth}
{\includegraphics[width=1.3\textwidth]{figures/BG505_prefs.pdf}}
\caption{\label{fig:BG505prefs}
Amino-acid preferences for the BG505 Env.
At each site, the height of the letter is proportional for that site's preference for that amino acid.
The top color bar indicates the region of Env (gp120 variable loop, gp120 not variable loop, or gp41).
The lower color bar indicates the evidence that the site evolves faster ($\omega_r > 1$) or slower ($\omega_r < 1$) than expected given the experiments~\citep[see][]{bloom2017identification}.
The letters above the logos indicate the wildtype amino acid in BG505.
Sites are numbered using the HXB2 scheme~\citep{korber1998numbering}).
This logo plot shows the site-specific amino-acid preferences for BG505 after averaging the replicates and re-scaling by the stringency parameter in \TABLE{phydms}.
The figure was generated using \texttt{dms\_tools2}~\citep[\url{https://jbloomlab.github.io/dms_tools2/}]{bloom2015software}, which in turn utilizes \texttt{weblogo}~\citep{crooks2004weblogo}.
}
\figdata{The numerical values of the amino-acid preferences plotted in this figure are in \texttt{rescaled\_BG505\_prefs.csv.}}
\figdata{The sequence of BG505 Env and mapping from sequential (\emph{original} column) to HXB2 numbering (\emph{new} column) is in \texttt{BG505\_to\_HXB2.csv}.}
\figdata{The $\omega_r$ values and associated $P$-values for BG505 in HXB2 numbering are in \texttt{BG505\_omegabysite.tsv}.}
\end{fullwidth}
\end{figure}

\begin{figure}
\begin{fullwidth}
{\includegraphics[width=1.3\textwidth]{figures/BF520_prefs.pdf}}
\caption{\label{fig:BF520prefs}
Amino-acid preferences for the BF520 Env.
This figure is the same as \FIG{BG505prefs} except that it shows the data for BF520 instead of BG505.
}
\figdata{The numerical values of the amino-acid preferences plotted in this figure are in \texttt{rescaled\_BF520\_prefs.csv.}}
\figdata{The sequence of BF520 Env and mapping from sequential (\emph{original} column) to HXB2 numbering (\emph{new} column) is in \texttt{BF520\_to\_HXB2.csv}.}
\figdata{The $\omega_r$ values and associated $P$-values for BF520 in HXB2 numbering are in \texttt{BG505\_omegabysite.tsv}.}
\end{fullwidth}
\end{figure}

A more qualitative way to assess if the deep mutational scanning authentically describes selection on Env function is to visually compare the measurements with existing knowledge.
\FIG{BG505prefs} and \FIG{BF520prefs} show the across-replicate average of the amino-acid preferences for each Env after re-scaling by the stringency parameters in \TABLE{phydms}.
(Note that throughout the rest of this paper, we use preferences re-scaled by these stringency parameters.)
At sites of known functional importance, these preferences are usually consistent with prior knowledge.
For instance, residues T257, D368, E370, W427, and D457 are important for Env binding to CD4~\citep{olshevsky1990identification}, and all these amino acids are highly preferred in our deep mutational scanning (\FIG{BG505prefs} and \FIG{BF520prefs}).
Likewise, Env has 10 disulfide bonds (linking sites 54-74, 119-205, 126-196, 131-157, 218-247, 228-239, 296-331, 378-445, 385-418, and 598-604), most of which are important for function~\citep{van2008only} -- and the cysteines at these sites are highly preferred in our deep mutational scanning.
The deep mutational scanning is also consistent with prior knowledge about sites that are tolerant of mutations.
For instance, Env has five variable loops that mostly evolve under weak constraint in nature~\citep{starcich1986identification,zolla2010structure} -- and most sites in these loops are mutationally tolerant in our deep mutational scanning (see sites indicated by gray overlay bars in \FIG{BG505prefs} and \FIG{BF520prefs}, such as 132 to 195).
It is beyond the scope of this paper to catalog associations between our measurements and all other prior mutational studies of Env, but the concordance of our findings with the above mutational studies, and the fact that our data improve phylogenetic models of Env's natural evolution, suggest that our measurements authentically describe functional selection on Env.

\subsection{Shifts in amino-acid preferences between BG505 and BF520}
The most fundamental question that we seek to address is how similar the amino-acid preferences are between the two Envs.
We have already noted that \FIG{mutfreqs}B shows that the preferences are more correlated for replicate measurements on the same Env than for replicate measurements on different Envs.
However, simply comparing correlation coefficients does not identify specific sites where mutational effects have shifted, nor does it quantify the magnitude of any shifts.

\begin{figure}
\begin{fullwidth}
{\bf \Large A} \hspace{0.61\textwidth} {\bf \Large B} \\
\hspace{0.01\textwidth} 
\includegraphics[width=0.58\textwidth]{figures/prefs_distance/prefs_distance.pdf}
\hspace{0.03\textwidth}
\includegraphics[clip=true, trim=0in 0in 0in 1.2in, width=0.68\textwidth]{figures/distance_distribution.pdf}
\vspace{0.15in}
\\ 
{\bf \Large C} \\
\includegraphics[width=1.3\textwidth]{figures/shifted_sites.pdf}
\caption{\label{fig:prefsdist}
Env sites with shifted amino-acid preferences between BG505 and BF520.
{\bf (A)} Calculation of the corrected distance between the amino-acid preferences of BG505 and BF520 at four example sites.
We have triplicate measurements for each Env.
We calculate the distance between each pair of replicate measurements, and group these into comparisons \emph{between} the two Envs and \emph{within} replicates for the same Env.
We compute the root-mean-square distance (RMSD) for both sets of comparisons, which we denote as RMSD$_{\rm{between}}$ and RMSD$_{\rm{within}}$.
The latter quantity is a measure of experimental noise.
The noise-corrected distance between Envs at a site, RMSD$_{\rm{corrected}}$, is simply the distance between the two Envs minus this noise.
{\bf (B)} The bottom distribution (orange) shows the corrected distances between BG505 and BF520 at all alignable sites.
The next distribution (blue) is a null generated by computing the corrected distances on all randomizations of the replicates among Envs.
The top two distributions (green) compare Env to the non-homologous influenza hemagglutinin (HA) protein~\citep{doud2016accurate} simply putting sites into correspondence based on sequence number.
We compute the $P$-value that a site has shifted between BG505 and BF520 as the fraction of the null distribution that exceeds that shift, and identify significant shifts at a false discovery rate (FDR) of 0.1 using the method of \citet{benjamini1995controlling}.
Using this approach, 30 of the 659 sites have significant shifts (corrected distance $\ge$0.22).
{\bf (C)} All sites that have significantly shifted their amino-acid preferences at an FDR of 0.01.
For each site, the logo stacks show the across-replicate average preferences for BG505 and BF520.
The sites are sorted by the magnitude of the shift.
}
\figdata{\label{figdata:prefsdist}The corrected distances between BG505 and BF520 at each site are in \texttt{BG505\_to\_BF520\_prefs\_dist.csv}.}
\end{fullwidth}
\end{figure}

We therefore used a more rigorous approach to identify sites where the amino-acid preferences differ between BG505 and BF520 by an amount that exceeds the noise in our experiments.
We first re-scaled the preferences from each experimental replicate by the stringency parameter for that Env from \TABLE{phydms} to calibrate all measurements to the stringency of natural selection.
We then identified the 659 sites in the mutagenized regions of Env that are pairwise alignable between BG505 and BF520 (\FIG{prefsdist}-source~data~\ref{figdata:prefsdist}).
For each site, we calculated the shift in amino-acid preferences between Envs using an approach similar to that of \citet{doud2015site} as illustrated in \FIG{prefsdist}A.
This approach calculates the magnitude of the shift after correcting for experimental noise by comparing the differences in preferences between replicates for BG505 and BF520 to the differences between replicates for the same Env.
\FIG{prefsdist}A shows this calculation for a site that has not shifted (site 598, which strongly prefers cysteine in both Envs), the most shifted site (512, which shifts from being mutationally tolerant in BG505 to strongly preferring alanine in BF520), and two other sites with more intermediate behaviors.

The overall distribution of shifts between BG505 and BF520 is shown in \FIG{prefsdist}B.
Most sites have relatively small shifts (close to zero), although there is a long tail of sites with large shifts.
This tail reaches its upper value with site 512, which has a shift of 0.52 out of a maximum possible of 1.0.
How should we interpret this distribution -- have mutational effects shifted a lot, or not very much?
We can establish an upper-bound for how much sites might shift by comparing Env to a \emph{non}-homologous protein.
\FIG{prefsdist}B shows the distribution of shifts when comparing Env to influenza's hemagglutinin protein, which has previously had its amino-acid preferences measured by deep mutational scanning~\citep{doud2016accurate} .
Most sites have large shifts between Env and hemagglutinin, with the typical shift being $\sim$0.4 and some approaching the maximum value of 1.0.
We can also establish a lower-bound by creating a null distribution for the expected shifts if all differences are simply due to experimental noise.
This null distribution is created by randomizing the experimental replicates among Envs.
\FIG{prefsdist}B shows that the null distribution is more peaked at zero than the real distribution, and does not have the same prominent tail of sites with large shifts.
The answer to the question of how much mutational effects have shifted is therefore nuanced: they have substantially shifted at some sites, but remain vastly more similar between the two Envs than between two unrelated proteins. 

We can use the null distribution to identify sites where the shifts between BG505 and BF520 are significantly larger than the noise in our experiments (\FIG{prefsdist}B).
There are 30 such sites at a false discovery rate of 0.1.
\FIG{prefsdist}C shows the amino-acid preferences of these significantly shifted sites for each Env.
For the majority of shifted sites, one Env prefers a specific amino acid whereas the other Env tolerates many amino acids; for instance, see sites 512, 516, 599, 165, 605 and 505 in \FIG{prefsdist}C.
Such broadening and narrowing of a site's mutational tolerance is frequently linked to changes in protein stability, with a more stable protein typically being more mutationally tolerant~\citep{wang2002evolution,bloom2006protein,gong2013stability,kumar2017stability}.
Work with engineered Env protein in the form of ``SOSIP'' trimer~\citep{binley2000recombinant,sanders2002stabilization} has shown that BG505 SOSIP is more thermostable than BF520 SOSIP~\citep{verkerke2016epitope}.
Consistent with this fact, sites with altered mutational tolerance are often (although not always, see sites 165 and 520 in \FIG{prefsdist}C) more mutationally tolerant in BG505.

However, not all of the significantly shifted sites show a simple pattern of broadening or narrowing of mutational tolerance.
For instance, site 288 does not alter its mutational tolerance but rather flips its rather narrow amino-acid preference from phenylalanine in BG505 to leucine in BF520 (\FIG{prefsdist}C).
Thus, there is variation in both the extent and types of shifts observed.

\subsection{Structural and evolutionary properties of shifted sites}

\begin{figure}
{\bf \Large A} \\
\includegraphics[width=0.8\textwidth]{figures/shifts_on_structure/shifts_on_structure.pdf}
\\
{\bf \Large B} \hspace{0.28\textwidth} {\bf \Large C} \hspace{0.28\textwidth} {\bf \Large D} \\
\includegraphics[width=0.28\textwidth]{figures/rsa_vs_shifts.pdf}
\hspace{0.01\textwidth}
\includegraphics[width=0.28\textwidth]{figures/shifts_proximity.pdf}
\hspace{0.01\textwidth}
\includegraphics[width=0.4\textwidth]{figures/conservation_vs_shifts.pdf} 
\caption{\label{fig:shiftfeatures}
Characteristics of significantly shifted sites.
{\bf (A)} One monomer of the Env trimer~\citep[PDB 5FYL;][]{stewart2016trimeric} is colored from gray to orange according to the magnitude of the mutational shift at each site (orange indicates large shift).
Sites that are significantly shifted according to \FIG{prefsdist}B are in spheres, and all other sites are in cartoon representation.
{\bf (B)} 
There is no significant difference in the relative solvent accessibility of sites that have and have not undergone significant shifts.
The absolute solvent accessibility of each site was calculated using DSSP~\citep{kabsch1983dictionary} and normalized to a relative solvent accessibility using the absolute accessibilities from \citet{tien2013maximum}.
{\bf (C)} 
Sites of significant shifts are clustered in Env's structure.
Shown is the distance of each significantly shifted and not-shifted site to the closest other shifted site.
{\bf (D)}
Large mutational shifts are \emph{not} strongly enriched at sites that have substituted between BG505 and BF520, or at sites that contact sites that have substituted. 
The plot shows the magnitudes of the shifts among the 83 crystallographically resolved sites that have substituted between BG505 and BF520, the 234 non-substituted sites that physically contact a substitution in the Env structure (any non-hydrogen atom within 3.5\angstrom), and all other sites.
\FIGSUPP[shiftfeatures]{subs_shifts} shows that there is a modest borderline-significant tendency of significantly shifted sites to have substituted.
All panels only show the 577 sites that are resolved in the crystal structure.
Of the 30 significantly shifted sites in \FIG{prefsdist}B, only 28 are crystallographically resolved.
Structural distances and solvent accessibilities were calculated using all monomers in the trimer.
$P$ values were calculated using the Mann-Whitney $U$ test.
}
\figsupp[\label{figsupp:subs_shifts}Statistical testing of whether significantly shifted sites are more likely to have substituted between BG505 and BF520.]
{The sites of significant shifts as determined in \FIG{prefsdist}B are somewhat more likely to be among the sites that have substituted between BG505 and BF520.
However, this association is only borderline statistically significant, with $P = 0.055$ using a Fisher's exact test.}
{\input{figures/shifts_vs_subs_table}}
\end{figure}

What distinguishes the sites that have undergone significant shifts?
\FIG{shiftfeatures}A shows the locations of the shifted sites on the crystal structure of Env.
There is no visually obvious tendency for shifted sites to preferentially be on Env's surface or in its core, and a statistical analysis (\FIG{shiftfeatures}B) finds no association between a site's relative solvent accessibility and whether its amino-acid preferences have shifted.
However, \FIG{shiftfeatures}A does suggest that the sites of significant shifts tend to cluster in Env's structure, and a statistical analysis confirms that this is the case (\FIG{shiftfeatures}C).
Therefore, the factors that drive shifts in Env's mutational tolerance often affect physically interacting clusters of residues in a coordinated fashion.

An obvious hypothesis is that strongly shifted sites have substituted between BG505 and BF520, or physically contact such substitutions.
According to this hypothesis, substitutions would alter the local physicochemical environment of the substituted site and its neighbors, thereby shifting the amino-acid preferences of sites in the physical cluster.
But surprisingly, the typical magnitude of shifts is not significantly larger at sites that have substituted, or at sites that contact sites that have experienced substitutions (\FIG{shiftfeatures}C).
There is a borderline trend for the significantly shifted sites to be more likely to have substituted between BG505 and BF520 (\FIGSUPP[shiftfeatures]{subs_shifts}), but most shifted sites have not substituted (only 8 of the 30 shifted sites differ in amino-acid identity between the two Envs).
The lack of strong enrichment in shifts at substituted sites contrasts with previous protein-wide experimental~\citep{doud2015site} and simulation-based~\citep{pollock2012amino,shah2015contingency} studies of shifting amino-acid preferences, which found that shifts were dramatically more pronounced at sites of substitutions.
The difference may arise because these earlier studies examined proteins that are fairly conformationally static (absolutely so in the case of the simulations).
On the other hand, Env is extremely complex and conformationally dynamic~\citep{munro2014conformational,ozorowski2017open}, which may increase the opportunities for long-range epistasis to enable substitutions at one site to shift the amino-acid preferences of distant sites.

\subsection{Entrenchment of substitutions modestly contributes to mutational shifts}

\begin{figure}
\centerline{\includegraphics[clip=true, trim=0in 0in 0in 0.7in, width=0.68\textwidth]{figures/entrenchment.pdf}}
\caption{\label{fig:entrenchment}
Entrenchment of substitutions during Env evolution.
There are 12,521 possible amino-acid mutations at the 659 mutagenized sites alignable between BG505 and BF520.
The blue densities show the effects of all these mutations to each Env.
The orange densities show the effects of just the 92 mutations that convert BG505 to BF520 or vice versa.
In the absence of entrenchment, mutating a site in BG505 to its identity in BF520 should have the opposite effect of mutating the site in BF520 to its identity in BG505.
In this case, we would expect the BF520$\rightarrow$BG505 distribution to be the mirror image of the BG505$\rightarrow$BF520 distribution -- and both distributions should be centered around zero if the two Envs are equivalently functional.
Instead, mutating a site in either Env to its identity in the other Env tends to be deleterious, indicating that substitutions are often entrenched in the Env in which they have fixed.
The effect of a mutation is quantified as the log of the ratio of the site's preference for the mutant amino acid to the preference for the wildtype amino acid.
}
\end{figure}

One idea that has recently gained support in the protein-evolution field is that substitutions become ``entrenched'' by subsequent evolution~\citep{pollock2012amino,shah2015contingency,starr2017pervasive}.
Entrenchment is the tendency of a mutational reversion to become increasingly unfavorable as a sequence evolves.
Given two homologs, if there is no entrenchment then the effect of mutating a site in the first homolog to its identity in the second will simply be the opposite of mutating the site in the second homolog to its identity in the first.
But if there is entrenchment, then both mutations will be unfavorable, since the site is entrenched at its preferred identity in each homolog.

\FIG{entrenchment} shows the distribution of effects for mutating all sites that differ between BG505 and BF520 to the identity in the other Env.
As expected under entrenchment, the average effect of these mutations is deleterious -- although there are a substantial number of sites where the mutational flips are not deleterious. 
We can get some sense of the magnitude of the entrenchment by comparing the effects of the BG505$\leftrightarrow$BF520 mutations to the distribution of effects of all possible amino-acid mutations (\FIG{entrenchment}).
This comparison shows that even unfavorable inter-Env mutational flips are generally more favorable than random amino-acid mutations.
Therefore, entrenchment occurs for some but not all substitutions that distinguish BG505 and BF520, and the magnitude of entrenchment is less than the effect of a typical random mutation.
Entrenchment of substitutions therefore contributes to some of the mutational shifts.
But given that many of these shifts occur at sites that do not even differ between the Envs (\FIG{shiftfeatures}D), entrenchment of substitutions is clearly not the only cause of the shifting amino-acid preferences.


\subsection{Comparing selection in the lab to natural selection}
Our experiments measure the effects of mutations on viral growth in a T-cell line in the lab.
But HIV actually evolves in humans, where additional selection pressures on Env are undoubtedly present.
For instance, antibody pressure might increase the rate of evolution at some sites~\citep{albert1990rapid,wei2003antibody,richman2003rapid}, whereas pressure to mask certain epitopes~\citep{kwong2002hiv} might add constraint at other sites. 
Comparing selection in our experiments to natural selection can identify sites that are under such additional pressures during HIV's actual evolution in humans.

\begin{figure}
\includegraphics[width=1.0\textwidth]{figures/omegabysite_structural_analysis/omegabysite_structural_analysis.pdf}
\caption{\label{fig:divsel}
Sites in Env that evolve faster or slower in nature than expected given the functional constraints measured in the lab.
We calculated the statistical evidence that each site evolves faster ($\omega_r > 1$) or slower ($\omega_r < 1$) than expected given the experimentally measured amino-acid preferences using the method of \citet{bloom2017identification}.
{\bf (A)}
One monomer of the Env trimer~\citep[PDB 5FYL;][]{stewart2016trimeric} is colored from blue to gray to red based on the strength of evidence that sites evolve slower than expected (blue), as expected (gray) or faster than expected (red) given the BG505 experiments.
Sites where the rate of evolution is significantly different than expected at a false discovery rate of 0.05 are shown in spheres.
{\bf (B)}
Like (A) but using the data from the BF520 experiments.
For both Envs, sites that evolve significantly slower or faster than expected are often on Env's surface (\FIGSUPP[divsel]{divselRSA}).
{\bf (C)}
The results are similar regardless of whether the BG505 or BF520 experiments are used.
Many of the sites of slower-than-expected evolution are asparagines in N-linked glycosylation motifs (\FIGSUPP[divsel]{glycans}).
All sites that evolve slower than expected for both experimental datasets are in \FIGSUPP[divsel]{slower}.
{\bf (D)} 
A large cluster of sites that evolve slower than expected is likely involved in Env's transition between open and closed conformational states.
Gray boxes indicate sites that \citet[PDB 5VN3]{ozorowski2017open} proposed form a hydrophobic network that regulates the conformational change; blue boxes and sticks indicate sites that evolve slower than expected.
All analyses used the phylogenetic tree in \FIG{tree}.
}
\figsupp[\label{figsupp:divselRSA}Relative solvent accessibilities of sites evolving faster or slower than expected.]{
Sites are grouped by whether they have $\omega_r > 1$ (diversifying selection) or $\omega_r < 1$ (purifying selection) at $Q < 0.05$ in \emph{both} Envs, or whether they fall into neither of these categories.
Relative solvent accessibilities were calculated as in \FIG{shiftfeatures}.
As can be seen from these box plots with overlaid points for each site, sites of both diversifying and purifying selection tend to have higher relative solvent accessibility than other sites.
}{\includegraphics[width=0.5\textwidth]{figures/selsites_rsa.pdf}}
\figsupp[\label{figsupp:glycans}Amino-acid preferences and alignment frequencies for glycosylation motifs.]
{This plot shows all sites that are N-linked glycosylation motifs in both BG505 and BF520.
Each motif is named by the residue number of the asparagine, and the preferences (averaged across replicates) for each Env are shown as well as the frequencies of amino acids across the clade A alignment at each of the three positions in the motif (N-X-S/T).
As can be seen from this plot, the experiments measure relatively broad mutational tolerance at many sites where the natural Env sequences have a strongly conserved motif.
We suspect this is because many glycans serve as a shield against immunity in nature -- a function that is not required in cell culture.}
{\includegraphics[width=\textwidth]{figures/glycan_prefs.pdf}}
\figsupp[\label{figsupp:slower}Amino-acid preferences and alignment frequencies of sites that evolve slower than expected.]{
This plot shows all sites that are evolving more slowly than expected in natural sequences given the preferences measured in both Envs.
Specifically, it shows all sites with $\omega_r < 1$ at $Q < 0.05$ for the ExpCMs for \emph{both} BG505 and BF520.
For each site, the plots show the preferences averaged across replicates and re-scaled for each Env, as well as the frequencies of amino acids in the clade A Env alignment. 
The $Q$-value indicated is the maximum of that for the two Envs.
}{\includegraphics[width=\textwidth]{figures/purifying_selsites.pdf}}
\figdata{The $\omega_r$ and $Q$-values are in \texttt{merged\_omegabysite.csv}.}
\end{figure}

We determined whether each site in Env evolves faster or slower in nature than expected given three models: that evolution is purely neutral (all nonsynonymous and synonymous mutations have equivalent effects), that sites are under the protein-level constraint measured in our experiments with BG505, or that sites are under the constraint measured with BF520. 
The first model used a standard dN/dS test~\citep[the ``FEL'' method of][]{kosakovsky2005not}, whereas the other two models performed a conceptually similar test but used ExpCM's that incorporate experimental measurements as described by \citet{bloom2017identification}.
The standard dN/dS model finds hundreds of sites that evolve slower than expected under neutral evolution (\TABLE{phydms}, $\omega_r < 1$), and only a handful of sites that evolve faster than expected under neutral evolution (\TABLE{phydms}, $\omega_r > 1$).
This finding is unsurprising, since it is well known that Env is under functional constraint.
In contrast, ExpCM's that test the rates of evolution relative to the experimentally measured constraints find far fewer sites that evolve slower than expected, but many more sites that evolve faster (\TABLE{phydms}). 

The sites that evolve slower or faster than expected from the experiments are shown in \FIG{divsel}A,B, and overlaid on the logoplots in \FIG{BG505prefs} and \FIG{BF520prefs} as the $\omega_r$ values.
The identified sites are similar regardless of whether we use the experiments with BG505 or BF520 (\FIG{divsel}C).
The reason the results are similar for both experimental datasets is that (as discussed above) the amino-acid preferences of \emph{most} sites are similar in both Envs, suggesting that either dataset provides a reasonable approximation of the site-specific functional constraints across the clade A Envs in \FIG{tree}.

What causes some sites to evolve faster or slower in nature than expected from the experiments?
The answer in both cases is likely to be immune selection.
Most of the sites of faster-than-expected evolution are on the surface of Env (\FIG{divsel}A,B and \FIGSUPP[divsel]{divselRSA}).
Env's escape from autologous neutralizing antibodies often involves amino-acid substitutions in surface-exposed regions~\citep{moore2009specificity}, including at many of the sites that evolve faster than expected.
Since our deep mutational scanning did not impose antibody pressure, sites where substitutions are antibody-driven will evolve faster in nature than expected from the experiments.

Interestingly, immune selection also offers a plausible explanation for the sites that evolve \emph{slower} than expected.
In addition to escaping immunity via substitutions at antibody-binding footprints, Env is notorious for employing a range of more general strategies to reduce its susceptibility to antibodies.
These strategies include shielding immunogenic regions with glycans~\citep{wei2003antibody,stewart2016trimeric,gristick2016natively} or hiding them by adopting a closed protein conformation~\citep{kwong2002hiv,guttman2015antibody,ozorowski2017open}.
Sites that contribute to such general immune-evasion strategies will be under a constraint in nature that is not present in our experiments -- and indeed, such sites evolve more slowly than expected from our experiments.
For instance, we find very little selection to maintain most glycans in our cell-culture experiments.
Of the 21 N-linked glycosylation sites shared between BG505 and BF520, only four are under strong selection to maintain the glycan in our experiments -- despite the fact that most are conserved in nature (\FIG{divsel}C and \FIGSUPP[divsel]{glycans}). 
This finding concords with prior literature suggesting that these glycans are selected primarily for their role in immune evasion~\citep{pugach2004prolonged,wang2013systematic,rathore2017glycosylation}. 
Similarly, a network of sites that help regulate Env's transition between open and closed conformations that have different antibody susceptibilities (\FIG{divsel}D) also evolve slower in nature than expected from our experiments.
Therefore, we can distinguish evolutionary patterns that are shaped by simple selection for Env function from those that are due to the additional complex pressures imposed during human infections. 

\section{Discussion}
We have experimentally measured the preference for each amino acid at each site in the ectodomain and transmembrane domain of two Envs under selection for viral growth in cell culture.
These amino-acid preference maps are generally consistent with prior knowledge about sites that are important for protein properties such as receptor binding or disulfide-mediated stability.
However, the main value of these maps comes not from comparing them with prior knowledge, but from the fact that such prior knowledge encompasses just a small fraction of the vast mutational space available to Env.
Because Env evolves so rapidly, every study of this protein must be placed in an evolutionary context, and our comprehensive amino-acid preference maps potentially enable this in ways that prior piecemeal studies of mutations cannot.

But these maps come with a potentially serious caveat: each one is measured for just a single Env variant.
The major question that our study aimed to answer is whether the maps are still useful for evolutionary questions, or whether Env's amino-acid preferences shift so rapidly that each map only applies to the specific HIV strain for which it was measured.
This question is reminiscent of one that was grappled with in the early days of protein crystallography, when it first became possible to build maps of a protein's structure.
Because it was not (and is still not) possible to crystallize every variant of a protein, it was necessary to determine whether protein structures could be usefully generalized among homologs.
Fortunately for the utility of structural biology, it soon became apparent that closely homologous proteins have similar structures~\citep{chothia1986relation,sander1991database}.
This rough generalizability of protein structures holds even for a protein as conformationally complex as Env -- for although there are many examples of mutations that alter aspects of Env's conformation and dynamics~\citep{kwong2000structures,white2010molecular,almond2010structural,davenport2013isolate}, SOSIP trimer structures from diverse Env strains remain highly similar in most respects~\citep{julien2015design,pugach2015native,stewart2016trimeric,verkerke2016epitope,gristick2016natively}.

Our results show that amino-acid preference maps of Env also have a useful level of conservation for many purposes.
From a qualitative perspective, the amino-acid preferences look mostly similar between BG505 and BF520, and so provide a valuable reference for estimating which mutations are likely to be tolerated at each site in diverse HIV strains.
Indeed, we anticipate that the complete maps of mutational effects in \FIG{BG505prefs} and \FIG{BF520prefs} will be useful for future sequence-structure-function studies.
From an analytical perspective, a powerful use of our maps is to identify sites that evolve differently in nature than is required by the simple selection for viral growth imposed in our experiments -- and the identified sites are largely the same regardless of whether the analysis uses an amino-acid preference map from BG505 or BF520.

Of course, from the perspective of protein evolution, the most interesting sites are the exceptions to the general conservation of amino-acid preferences.
Consistent with studies of other proteins~\citep{natarajan2013epistasis,harms2014historical,doud2015site,starr2017pervasive}, we find a subset of sites that change markedly in which mutations they tolerate.
Some shifted sites simply accommodate more amino acids in the more stable BG505 Env -- a type of shift that has been well-documented for other proteins~\citep{wang2002evolution,bloom2006protein,gong2013stability,kumar2017stability}.
But interestingly, there is no strong trend for shifts to be enhanced at sites that differ between BG505 and BF520.
Recent studies of protein evolution have focused on the idea that substitutions become ``entrenched'' as sites shift to accommodate new amino acids~\citep{pollock2012amino,shah2015contingency,bazykin2015changing,starr2017pervasive}.
Indeed, a prior protein-wide comparison of amino-acid preferences across homologs of influenza nucleoprotein found a significant enrichment of shifts at sites of substitutions~\citep{doud2015site}.
But although there is some entrenchment of differences between BG505 and BF520, this is not the major factor behind the shifts in amino-acid preferences: most sites that have shifted between BG505 and BF520 actually have the same wildtype amino acid in both Envs.
This rather surprising result might be due to Env's exceptional conformational complexity -- mutations can cause long-range alterations in Env's conformation~\citep{kwong2000structures,white2010molecular,almond2010structural,davenport2013isolate}, so it seems plausible that they might also shift mutational tolerance at distant sites. 

Our experiments provide highly quantitative data on the mutational tolerance of Env under selection for viral growth in cell culture.
These data are amenable to rigorous functional and evolutionary analyses.
Here we have shown how these data can be compared between Envs to identify sites where mutational tolerance shifts with viral genotype, or between experiments and nature to identify sites under different pressure in the lab and in humans.
Future experiments that modulate selection pressures in other relevant ways should provide further insight into the forces that drive and constrain HIV's evolution.

\clearpage
\section{Methods and Materials}

\subsection{Creation of codon-mutant libraries}
Our codon mutant libraries mutagenized all sites in \textit{env} to all 64 codons, except that the signal peptide and cytoplasmic tail were not mutagenized.
The rationale for excluding these regions is that they are not part of Env's ectodomain, and may be prone to mutations that strongly modulate Env's expression level.

The codon-mutant libraries were generated using the approach originally described in \citet{bloom2014experimentally}, with the modification of \citet{dingens2017comprehensive} to ensure more uniform primer melting temperatures.
The computer script used to design the mutagenesis primers (along with some detailed implementation notes) is at \url{https://github.com/jbloomlab/CodonTilingPrimers}.
For BF520, the three libraries are the same ones described by \citet{dingens2017comprehensive}.
For BG505, we created three libraries for this study.
The wildtype BG505 sequence used for these libraries is in Supplemental~file~\ref{suppfile:BG505}.
The BG505 mutagenesis primers are in Supplemental~file~\ref{suppfile:BG505primers}.

The end primers for the BG505 mutagenesis were: \seqsplit{5'-tgaaggcaaaactactggtccgtctcgagcagaagacagtggcaatgaga-3'} and \seqsplit{5'-gctacaaatgcatataacagcgtctcattctttccctaacctcaggcca-3'}.
As with BF520, we cloned the BG505 \textit{env} libraries into the \textit{env} locus of the full-length proviral genome of HIV strain Q23 \citep[another subtype-A transmitted/founder virus;][]{poss1999variants} using the high-efficiency cloning vector described in \citet{dingens2017comprehensive}.
For this cloning, we digested the cloning vector with BsmBI, and then used PCR to elongate the amplicons to include 30 nucleotides at each end that were identical in sequence to the ends of the BsmBI-digested vector.
The primers for this PCR were:
\seqsplit{5'-agataggttaattgagagaataagagaaagagcagaagacagtggcaatgagagtgatgg-3'} and \seqsplit{5'-ctcctggtgctgctggaggggcacgtctcattctttccctaacctcaggccatcc-3'}.
Next, we used NEBuilder HiFi DNA Assembly (NEB, E2621S) to clone the \textit{env} amplicons into the BsmBI-digested plasmids.
We purified the assembled products using Agencourt AMPure XP beads (Beckman Coulter, A63880) using a bead-to-sample ratio of 1.5, and then transformed the purified products into Stellar electrocompetent cells (Takara, 636765).
The transformations yielded between 1.5-3.6 million unique clones for each of the three replicate libraries, as estimated by plating 1:2,000 dilutions of the transformations.
We scraped the plated colonies and maxiprepped the plasmid DNA; unlike in \citet{dingens2017comprehensive}, we did not include a 4-hour outgrowth step after the scraping step.
For the wildtype controls, we maxiprepped three independent cultures of wildtype BG505 \textit{env} cloned into the same Q23 proviral plasmid.
See \FIGSUPP[dms_schematic]{sanger} and \FIG{mutfreqs}A for information on the average mutation rate in these libraries as estimated by Sanger sequencing and deep sequencing, respectively.

\subsection{Generation and passaging of viruses}
For BG505, we generated mutant virus libraries from the proviral plasmid libraries by transfecting 293T cells in three 6-well plates (so 18 wells total per library) with a per-well mixture of 2 $\mu$g plasmid DNA, 6 $\mu$l FuGENE 6 Transfection Reagent (Promega, E269A), and 100 $\mu$l DMEM.
The 293T cells were seeded at $5\times 10^5$ cells/well in D10 media (DMEM supplemented with 10\% FBS, 1\% 200 mM L-glutamine, and 1\% of a solution of 10,000 units/mL penicillin and 10,000 $\mu$g/mL streptomycin) the day before transfection, such that they were approximately 50\% confluent at the time of transfection.
In parallel, we generated wildtype viruses by transfecting one 6-well plate of 293T cells with each wildtype plasmid replicate.
At 2 days post-transfection, we harvested the transfection supernatant, passed it through a 0.2 $\mu$m filter to remove cells, treated the supernatant with DNAse to digest residual plasmid DNA as in~\citet{haddox2016experimental}, and froze aliquots at -80$^{\circ}$C.
We thawed and titered aliquots using the TZM-bl assay in the presence of 10 $\mu$g/mL DEAE-dextran as described in~\citet{dingens2017comprehensive}.

We conducted the low MOI viral passage illustrated in \FIG{dms_schematic}A in SupT1.CCR5 cells~\citep[obtained from Dr. James Hoxie;][]{boyd2015mutations}.
During this passage, cells were maintained in R10 media, which has the same composition as the D10 described above, except RPMI-1640 (GE Healthcare Life Sciences, SH30255.01) is used in the place of DMEM.
In addition, the media contained 10 $\mu$g/mL DEAE-dextran to enhance viral infection.
We infected cells with 4 million (for replicate 1) or 5 million (for replicates 2 and 3) TZM-bl infectious units of mutant virus at an MOI of 0.01, with cells at a starting concentration of 1 million cells/mL in vented tissue-culture flasks (Fisher Scientific, 14-826-80).
At day 1 post-infection, we pelleted cells, aspirated the supernatant, and resuspended cell pellets in the same volume of fresh media still including the DEAE-dextran.
At 2 days post-infection, we doubled the volume of each culture with fresh media still including DEAE-dextran.
At 4 days post-infection, we pelleted cells, passed the viral supernatant through a 0.2 $\mu$m filter, concentrated the virus $\sim$30 fold using ultracentrifugation as described in \citet{dingens2017comprehensive}, and froze aliquots at -80$^{\circ}$C.
In parallel, for each replicate, we also passaged $2\times 10^5$ (for replicate 1) or $5 \times 10^5$ (for replicates 2 and 3) TZM-bl infectious units of wildtype virus using the same procedure.
To obtain final titers for our concentrated virus, we thawed one of the aliquots stored at -80$^{\circ}$C and titered using the TZM-bl assay in the presence of 10 $\mu$g/mL DEAE-dextran.

For the final short-duration infection illustrated in \FIG{dms_schematic}A, for each replicate we infected $10^6$ TZM-bl infectious units into $10^6$ SupT1.CCR5 cells in the presence of 100 $\mu$g/mL DEAE-dextran (note that this is a 10-fold higher concentration of DEAE-dextran than for the other steps, meaning that the effective MOI of infection is higher if DEAE-dextran has the expected effect of enhancing viral infection).
Three hours post-infection, we pelleted the cells and resuspended them in fresh media without any DEAE-dextran.
At 12 hours post-infection, we pelleted cells, washed them once with PBS, and then used a miniprep kit to harvest reverse-transcribed unintegrated viral DNA \citep{haddox2016experimental}.

The generation, passaging and deep sequencing of BF520 was done in a highly similar fashion, except that we only had a single replicate of the wildtype control.
Note that the final passaged BF520 mutant libraries analyzed here actually correspond to the ``no-antibody'' controls described in \citet{dingens2017comprehensive}, but that study did not analyze the initial plasmid mutant libraries relative to these passaged viruses, and so was not able to provide measurements of the amino-acid preferences.

\subsection{Illumina deep sequencing}
We deep sequenced all of the samples shown in \FIG{mutfreqs}A: the plasmid mutant libraries and wildtype plasmid controls, and the cDNA from the final mutant viruses and wildtype virus controls.
In order to increase the sequence accuracy, we used a barcoded-subamplicon sequencing strategy.
This general strategy was originally applied in the context of deep mutational scanning by \citet{wu2014high}, and the specific protocol used in our work is described in \citet{doud2016accurate} (see also \url{https://jbloomlab.github.io/dms_tools2/bcsubamp.html}).

The primers used for BG505 are in Supplementary~file~\ref{suppfile:B505bcsubamp}.
The primers used for BF520 are in \citet{dingens2017comprehensive}.
The data generated by the Illumina deep sequencing are on the Sequence Read Archive under the accession numbers provided at the beginning of the Jupyter notebook in Supplementary~files~\ref{suppfile:code} and \ref{suppfile:html}.

\subsection{Analysis of deep-sequencing data}
We analyzed the deep-sequencing data using the \texttt{dms\_tools2} software package~\citep[\url{https://jbloomlab.github.io/dms_tools2/}]{bloom2015software}.
The algorithm that goes from the deep-sequencing counts to the amino-acid preferences is that described in \citet{bloom2015software} (see also \url{https://jbloomlab.github.io/dms_tools2/prefs.html}).
A Jupyter notebook that performs the entire analysis including generation of most of the figures in this paper is in Supplementary~file~\ref{suppfile:code}.
An HTML rendering of this notebook is in Supplementary~file~\ref{suppfile:html}.

The Jupyter notebooks in Supplementary~files~\ref{suppfile:code} and \ref{suppfile:html} also contain numerous plots that summarize relevant aspects of the deep sequencing such as read depth, per-codon mutation frequency, mutation types, etc.
Supplementary~file~\ref{suppfile:code} also contains text files and CSV files with the numerical values shown in these plots.

Citations are also owed to \texttt{weblogo}~\citep[][\url{http://weblogo.threeplusone.com/}]{crooks2004weblogo} and \texttt{ggseqlogo}~\citep[][\url{https://omarwagih.github.io/ggseqlogo/}]{wagih2017ggseqlogo}, which were used in the generation of the logoplots. 

\subsection{Alignments and phylogenetic analyses of Env sequences}
A basic description of the process used to generate the clade A sequence alignment in \FIG{tree}-source~data~\ref{figdata:alignment}, the alignment mask in \FIG{tree}-source~data~\ref{figdata:mask}, and the phylogenetic tree in \FIG{tree} are provided in the legend to that figure.
An algorithmic description of how the alignment and tree were generated are in Supplementary~files~\ref{suppfile:code} and \ref{suppfile:html}.

For fitting of the phylogenetic substitution models, we used \texttt{phydms}~\citep[][\url{http://jbloomlab.github.io/phydms/}]{hilton2017phydms} to optimize the substitution model parameters and branch lengths on the fixed tree topology in \FIG{tree}.
The Goldman-Yang (or YNGKP) model used in \TABLE{phydms} is the M5 variant described by \citet{yang2000codon}, with the equilibrium codon frequencies determined empirically using the CF3x4 method~\citep{pond2010correcting}.
For the ExpCM shown in \TABLE{phydms}, we extended the models with empirical nucleotide frequencies described in \citet{hilton2017phydms} to also allow $\omega$ to be drawn from discrete gamma-distributed categories exactly as for the M5 model.
These ExpCM with gamma-distributed $\omega$ were implemented in \texttt{phydms} using the equations provided by \citet{yang1994maximum} (see also \url{http://jbloomlab.github.io/phydms/implementation.html#models-with-a-gamma-distributed-model-parameter}).
The preferences were re-scaled by the stringency parameters in \TABLE{phydms} as described in \citet{hilton2017phydms}.
For both the M5 model and the ExpCM with a gamma-distributed $\omega$, we used four categories for the discretized gamma distribution.

\TABLE{phydms}-source-data~\ref{tabledata:phydms} shows the results for a wider set of models than those used in \TABLE{phydms}. 
These include the M0 model of \citet{yang2000codon}, ExpCM without a gamma-distributed $\omega$, and ExpCM in which the amino-acid preferences are averaged across sites as a control to ensure that the improved performance of these models is due to their site-specificity.
Note how for these Env alignments, using a gamma-distributed $\omega$ is very important in order for the ExpCMs to outperform the M5 model -- we suspect this is because there are many sites of strong diversifying selection.

For detection of sites with faster or slower than expected evolution, we used the approach in \citet{bloom2017identification}, which is exactly modeled on the FEL approach of \citet{kosakovsky2005not} but extended to ExpCM.
This approach estimates a $P$-value that $\omega_r$ is not equal to one for each site $r$ using a likelihood-ratio test.
For the Q-values and false discovery rate testing, we considered the tests for $\omega_r > 1$ and $\omega_r < 1$ separately.

Supplementary~files~\ref{suppfile:code} and \ref{suppfile:html} contains the code that runs \texttt{phydms} to reproduce all of these analyses.

\subsection{Identifying sites of shifted amino-acid preference}
When identifying shifts in amino-acid preferences between the two Envs, we needed a way to quantify differences between the Envs while accounting for the fact that our measurements are noisy.
The approach we use is based closely on that of \citet{doud2015site}, and is illustrated graphically in \FIG{prefsdist}A.
The RMSD$_{\rm{corrected}}$ value is our measure of the magnitude of the shift.
\FIG{prefsdist}A, its legend, and the associated text completely explains these calculations with the following exception: they do not detail how the ``distance'' between any two preference measurements was calculated. 
The distance between preferences at each site was simply defined as half of the sum of absolute value of the difference between preferences for each amino acid.
Specifically, for a given site $r$, let $\pi_{r,a}^{i}$ be the preference for amino-acid $a$ in homolog $i$ (e.g., BG505) and let $\pi_{r,a}^{j}$ be the preference for that same amino acid in homolog $j$ (e.g., BF520). 
Then the distance between the homologs at this site is simply $D_{r}^{i,j} = \frac{1}{2}\sum_{a}|\pi_{r,a}^{i}-\pi_{r,a}^{j}|$.
The factor of $\frac{1}{2}$ is used so that the maximum distance will always fall between zero and one.

\subsection{Analysis of entrenchment}
For the analysis in \FIG{entrenchment}, the results are presented in terms of the mutational effects rather than the amino-acid preferences. 
If $\pi_{r,a}$ is the preference of site $r$ for amino-acid $a$ and $\pi_{r,a'}$ is the preference for amino-acid $a'$ (both re-scaled by the stringency parameters in \TABLE{phydms}), then the estimated effect of the mutation from $a$ to $a'$ is simply $\log\left(\frac{\pi_{r,a'}}{\pi_{r,a}}\right)$.

\subsection{Data and code availability}
All code and input data required to reproduce all analyses in this paper are in Supplementary~file~\ref{suppfile:code} (see also Supplementary~file~\ref{suppfile:html}).
The deep sequencing data are on the Sequence Read Archive with the accession numbers listed in Supplementary~files~\ref{suppfile:code} and \ref{suppfile:html}.

\section{Acknowledgments}
We thank Michael Doud and Orr Ashenberg for computer code that formed the basis for some of the analyses.
We thank Andrew Ward for pointing out to us that some of the sites with slower-than-expected rates of evolution are involved in Env's conformational changes upon receptor binding.
We thank Kelly Lee for helpful input about the relative stabilities of the BG505 and BF520 Envs.
We thank the Fred Hutch Genomics Core for performing the Illumina deep sequencing.

\bibliography{references.bib}

\clearpage

\begin{suppfile}
\caption{
\label{suppfile:code}
The code to perform all steps in the analysis is in \texttt{analysis\_code.zip}.
Specifically, this file contains a Jupyter notebook that performs the analysis, all required input data, and all reasonably sized output files.
The Jupyter notebook downloads the deep sequencing data, processes it with the \texttt{dms\_tools2} software~\citep[\url{https://jbloomlab.github.io/dms_tools2/}]{bloom2015software}, and also performs a variety of downstream analyses that generate most of the figures for this paper.
}
\end{suppfile}

\begin{suppfile}
\caption{
\label{suppfile:html}
An HTML rendering of the Jupyter notebook that performs the computational analysis.
The actual notebook is in Supplementary~file~\ref{suppfile:code}, but if you just want to look at the analysis rather than run it, then you may prefer this file instead.
In particular, the notebook contains plots detailing the deep sequencing data analysis as generated using the \texttt{dms\_tools2} software~\citep[\url{https://jbloomlab.github.io/dms_tools2/}]{bloom2015software}.
}
\end{suppfile}

\begin{suppfile}
\caption{
\label{suppfile:BG505}
The sequence of the wildtype BG505 \textit{env} used in our study is in FASTA format in the file \texttt{BG505\_env.fasta}.
}
\end{suppfile}

\begin{suppfile}
\caption{
\label{suppfile:BG505primers}
The sequences of the primers used for the BG505 codon mutagenesis are in the file \texttt{BG505\_codon\_mutagenesis\_primers.txt}.
}
\end{suppfile}

\begin{suppfile}
\caption{
\label{suppfile:B505bcsubamp}
The primers used for the BG505 barcoded-subamplicon sequencing are in the file \texttt{BG505\_bcsubamp\_primers.txt}.
}
\end{suppfile}


\end{document}
