\documentclass[11pt, oneside]{article}   	% use "amsart" instead of "article" for AMSLaTeX format
\usepackage{geometry}                		% See geometry.pdf to learn the layout options. There are lots.
\geometry{letterpaper}                   		% ... or a4paper or a5paper or ... 
\usepackage{color}
\usepackage[parfill]{parskip}    		% Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}				% Use pdf, png, jpg, or epsÂ§ with pdflatex; use eps in DVI mode
								% TeX will automatically convert eps --> pdf in pdflatex	
								
\usepackage{hyperref}
	
\usepackage{amssymb}



\title{Response to reviews of ``Mapping mutational effects along the evolutionary landscape of HIV envelope'' for \textit{eLife}}
\author{Hugh K. Haddox, Adam S. Dingens, Sarah K. Hilton, Julie Overbaugh, and Jesse D. Bloom}

\begin{document}
\maketitle

\emph{Below, the original comments {\color{blue} are in blue}, and our responses are in black.}

\color{blue}

\section*{SUMMARY} 

Information on the possible amino acid trajectories of the rapidly evolving HIV envelope protein has long been limited. In this manuscript, you extend previous work from your lab analyzing the mutational landscape of the HIV-1 Env gene during in vitro viral replication. The work focuses on two clade A transmitter founder viruses, BG505 and BF520, which differ by more than 100 mutations. A deep mutational scanning approach is used to identify the permitted mutations at each residue of Env. Of particular interest is how amino acid preferences differ for these two proteins. A set of 30 sites with clear shifts in amino acid preference are identified. These data are compared to known mutational constraints on Env and to one another, and sources for the shifts in mutant affects are proposed by eliminating many scenarios. The results offer a body of knowledge useful for identifying i) residues and networks that are critical for basic function and ii) hot spots for evolvability or immune pressure. The analysis has been performed to a high standard (though some clarifications are necessary; see minor comments), and the openness of the data is a great service to the community. The paper represents a significant contribution towards understanding the structure-function relationship of HIV envelope. Therefore, the paper is likely to be accepted for publication if the following points are adequately addressed. 

{\color{black}
Thanks for the excellent summary of our work.
We appreciate the favorable evaluation of the rigor of the work and the importance of the data.
The comments below are extremely helpful, and we have revised the manuscript in line with these suggestions.
These revisions have further strengthened the manuscript.
}

\section*{ESSENTIAL REVISIONS} 

1. How are mutations that strongly modulate Env's expression level taken into account? 

{\color{red}
The best I can think of here is just to explain our rationale for excluding signal peptide and cytoplasmic tail. 
I don't think we have a way of addressing this comment beyond that?
Hugh and Adam, do you have additional ideas?
}

2. Only the native, pre-fusion Env trimer is used to calculate proximity of substituted and shifted residues. It is possible that substitution-shift colocalization may be detected in other conformations of Env as it undergoes structural rearrangement to mediate fusion, as opposed to acting through longer-range allostery in the pre-fusion conformation. Another well-characterized structure that should be considered is the post-fusion six-helix bundle of gp41. 

{\color{red}
This is an excellent suggestion. We investigated substitution-shift co-localization in two other conformations: the CD4-bound conformation and the post-fusion six-helix bundle. In the end, our basic conclusions about co-localization remained the same. But, we believe these other conformations were extremely important to consider, and that the results substantially strengthen the conclusion that epistasis may be acting via long-range interactions.

First, we investigated the CD4-bound conformation.

Note: In the process of making these revisions, we discovered a minor bug that affected our calculations of residue-residue distances. We have fixed that bug in the revised version. None of the qualitative results change (the trends that were previously significant are still significant, and the ones that previously were not significant are still not significant. However, some of the quantitative values in the box plots and the corresponding P-values have changed slightly.
}

3. Overall, it is found that the shifted sites tend to cluster together, but they are not necessarily the same as the sites with substitutions between the two Envs or ones that contact them. This analysis is not pursued in great detail, however, leaving the reasons why shifted sites tend to cluster together unexplained. Given that the authors also observe some evidence of entrenchment of substitutions between the Envs, is it possible that interactions between the shifted sites themselves are important? This question could be explored through a more detailed examination of the structure. 

{\color{red}
Seems like this might be best examined by looking carefully at some of the clusters?
I'm not sure how to do it with a statistical analysis.
Hugh and Adam---do you have any thoughts on this?
}

\section*{MINOR POINTS}

1. Readers may miss the rationale for excluding the signal peptide and cytoplasmic tail if it is only found in the Methods. 

{\color{red}Yes, we should move this up. Will be done in context of answering essential revision 1, and then we can describe change here.}

2. Page 3: When referring to the modifications made in this paper versus the previous publication, it is not clear whether the cell type difference was expression of CCR5 or use of a new cell line altogether. 

{\color{black}
We have clarified that we used the SupT1.CCR5 cell line to allow for infection with viruses bearing CCR5-tropic Envs. This cell line is a SupT1 clone that was lentiviral transduced to also express CCR5 in addition to CXCR4 and CD4, which SupT1 cells express naturally. In our original work with CXCR4-tropic LAI Env, we used SupT1 cells. 
}

3. Page 6: The dN/dS parameter is introduced, but its meaning in the two different models is only described in the next paragraph. Noting that a description follows may help readers. 

{\color{red}
Easy to fix.}

4. The color scheme of structures in Fig 7a, 9a is difficult to discern. Higher contrast colors please. 

{\color{red}
Hugh, can you do this?}

5. The methodology used here was previously developed in your group, which would benefit from a short description in the main text. For example, how exactly is $\omega$ defined, what is the stringency parameter, can the use of $\omega$ be motivated, how is $\omega_r$ defined and motivated etc? What are the assumptions in using these measures? We think this is necessary for a stand-alone paper aimed at a broad audience. All the details of the method should not be repeated, but an intuitive summary in the main text would be helpful. 

{\color{red}
This is easy to do. Maybe Jesse should do this?}

6. With the current version 2.2.0 of dms\_tools2 installed via pip today (January 19, 2018), the included iPython notebook throws an error. Error text is below: 
\begin{verbatim}
--------------------------------------------------------------------------- 
ModuleNotFoundError Traceback (most recent call last) 
in <module>() 
36 import dms_tools2.dssp 
37 import dms_tools2.compareprefs 
---> 38 import dms_tools2.protstruct 
39 import dms_tools2.rplot 
40 print(dms_tools2.rplot.versionInfo()) 

ModuleNotFoundError: No module named 'dms_tools2.protstruct'
\end{verbatim}

{\color{black}
This is now fixed.
Previously we were running the analysis with a developmental version of \texttt{dms\_tools2} (version 2.2.dev1) that was not yet on \texttt{PyPI}.
Now the analysis uses a stable version (2.2.4) that is on \texttt{PyPI} and installable with \texttt{pip}.
Therefore, the analysis should now work if \texttt{dms\_tools2} is installed with \texttt{rplot} as described here: \url{https://jbloomlab.github.io/dms_tools2/installation.html#installing-the-rplot-module}.
Note that some other external Python packages are also needed as indicated by the imports at the beginning of the iPython notebook.
}

7. It might be helpful to increase the transparency on the points in Figure 3B, because at the current level it's impossible to resolve any detail in the blob near zero. 

{\color{black}
We have added a new figure supplement (Figure 3--Figure supplement 2) that shows the correlations as a KDE contour plot rather than a scatter plot.
As is apparent from this new supplement, the vast majority of points are near the origin (low preference in both replicates). 
This is expected since most mutations are highly deleterious.
Therefore, the correlation signal is due to relatively small fraction of amino acids that have non-negligible preference---a fact that makes sesne, since most mutations are so deleterious as to be evolutionarily irrelevant.
Unfortunately, given the skew of the data it is not possible to find a transparency that strikes a good balance between showing the high-preference points clearly and not over-plotting the low preference points.
Therefore, we have elected to keep the original Figure 3B (scatter plot) while also adding the new contour plot in Figure 3--Figure supplement 2) to show both elements of the data.
}

8. In Figure 4, it's not clear why the p-value for $\omega$ greater/less than one is used for shading rather than the estimated $\omega$ itself. 

{\color{black}
We have added text in the legend of Figure 4 to explain this.
Briefly, for individual sites, the point estimates of dN/dS values (or $\omega_r$) are not reliable when using a FEL framework for the reasons described in Pond et al (2015) and Murrell et al (2012).

Here is the basic intuition. 
Imagine two sites: one has 3 nonsynonymous substitutions and 0 synonymous substitutions, while the other has 20 nonsynonymous substitutions and 2 synonymous substitutons.
If we just calculated $\omega_r$, we would get a value of infinity for the first site and 10 for the second site.
But in reality, we think there is substantially stronger evidence for diversifying selection at the second site, since there are many more nonsynonymous substitutions.
In an FEL framework, we assess the statistical evidence that $\omega_r \ne 1$, and find much more evidence for the second site.
This increased evidence is manifested by the larger $P$-value rather than the value of $\omega_r$ itself.}

9. What is the rationale for rescaling amino acid preferences by the stringency parameter before comparing them? Based on (Hilton et al., 2017), my understanding is that only the amino acid preferences are measured directly. The stringency parameter appears to be based on a phylogenetic model fit. This rescaling would emphasize differences in the basic ordering of amino acid preferences over differences of intensity, but the latter would seem to also be important. 

{\color{black}
In this comment and the one immediately below (\#10), the reviewers point out the need to both justify why the preferences were re-scaled, and to clearly explain that this was done.

In the Results section where we first discuss the re-scaling, we have now added an entire paragraph that discusses this fact:
\begin{quote}
\textsl{For the entire rest of the paper, we use the experimentally measured preferences re-scaled by the stringency parameters in Table 1.
The reason we do this is to distinguish genuine differences between the two Envs from mere variation in the strength of selection between the two sets of experiments.
Re-scaling both sets of preferences to optimally describe Env evolution in nature is a principled way to standardize the measurements; see Hilton et al (2017) and the Methods section entitled ``Re-scaling the preferences'' for a more detailed explanation.}
\end{quote}
At other key points (such as Figure 6 and the bottom of page 18 in the original manuscript) we now also emphasize the fact that the re-scaled preferences are used.

The detailed justification for re-scaling is provided in the new Methods section entitled ``Re-scaling the preferences.''
Briefly, the preferences themselves are calculated from the enrichment of mutations after selection.
But the extent of enrichment is a function of how the experiment was performed (e.g., how long the selection was performed) as well as the inherent effects of the mutations. 
For instance, let's say a given mutation is depleted 3-fold after 2 rounds of viral replication. 
Then if we instead allowed $2^2 = 4$ rounds of viral replication, then we would expect the mutation to be depleted $3^2 = 9$-fold.
Although we did our best to standardize the experiments with the two different Envs (BG505 and BF520), it is impossible to completely standardize experiments with different viruses since they grow at different inherent rates.
Therefore, we need some method to ensure that we have put the measurements for the two Envs on the same ``scale.''
Re-scaling by the stringency parameter is a principled way to do this.
See the new Methods section ``Re-scaling the preferences'' for a justification of why this is a principled approach.
}

10. In the text (p8, bottom) it is stated that the amino acid preferences are rescaled for comparison. But this is not mentioned in Figure 6 or described clearly in the Methods (p18, bottom). For the sake of clarity, it would be helpful to introduce new notation for rescaled preferences and indicate clearly where they are used. 

{\color{black}
This is an important point that we have addressed.
Se our response to comment \#9 immediately above for a description of the changes that we have made.}

11. In the discussion it is noted that: "But although there is some entrenchment of differences between BG505 and BF520, this is not the major factor behind the shifts in amino-acid preferences: most sites that have shifted between BG505 and BF520 actually have the same wildtype amino acid in both Envs." Does wildtype refer to the amino acids in the actual BG505 and BF520 sequences? The most preferred amino acid is the same in only 12/30 cases, based on Figure 6C. 

{\color{black}
This was unclear in the original manuscript.
We have updated Figure 6C to also show the wildtype amino acid at each site.
This makes it clear that even though the most preferred amino acid is only the same in 12/30 cases, the wildtype amino acid is the same in 18/30 sequences.
This is because although the wildtype is always among the most preferred amino acid, it is not always the most preferred one.
We have updated the sentence in the Discussion to explain this, and also to say the ``majority of sites'' rather than ``most sites'' since the former probably is a more accurate description of 18/30.}

12. A body of computational work that has been done to infer the effects of epistasis on HIV proteins (including ENV) is not adequately referenced. 
{\color{red}
This includes the new Arup Chakroborty paper. Others?}


\end{document}  